---
title: "phylosmith"
author: "Schuyler D. smith"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{phylosmith}
  %\VignetteEngine{knitr::rmarkdown}
  \usepackage[utf8]{inputenc}
---

```{r load_data, echo=FALSE, message=FALSE, warning=FALSE}
knitr::opts_chunk$set(fig.width=8, fig.height=4, output.lines=6)
library(phylosmith)
data(soil_column)
```

## `phylosmith` 

`phylosmith` is a conglomeration of functions written to process and analyze `phyloseq-class` objects. <a href="https://joey711.github.io/phyloseq/" targe="_blank" >Phyloseq</a> objects are a great data-standard for microbiome, gene-expression, and many other data types.

A lot of these functions are just to make "data-wrangling" easier for the user. Others will implement complex routines in a, hopefully, efficient and concise manner. I have also made functions to make figures for quick examination of data, but they may or may not be suitable for publication as some may require parameter optimization.

# Installation
```r
library(devtools)
install_github('schuyler-smith/phylosmith')
library(phylosmith)
```
\*for WINDOWS you need to install <a href="https://cran.r-project.org/bin/windows/Rtools/" target="_blank" >Rtools</a>, when prompted, select `add rtools to system PATH`.

# Functions
## Data Parsing
Call			 | Use
---------------- | ------------------------------------------------
[find_common_taxa](#find_common_taxa) | find taxa common to each treatment
[find_unique_taxa](#find_unique_taxa) | find taxa unique to each treatment
[merge_asvs](#merge_asvs)       | combine ASVs to lowest common biological sequence
[merge_samples](#merge_samples) | combines samples based on common factor within sample_data
[merge_treatments](#merge_treatments) | combines multiple columns in meta-data into a single column
[order_phyloseq_metadata](#order_phyloseq_metadata) | sets the orders of the factors in a sample_data column (for ordering graphs)
[relative_abundance](#relative_abundance) | transform abundance data to relative abundance
[taxa_filter](#taxa_filter) | filter taxa by proportion of samples seen in

## Graphs
Call			 | Use
---------------- | ------------------------------------------------
[abundance_heatmap_ggplot](#abundance_heatmap_ggplot) | create a ggplot object of the heatmaps of the abndance table
[abundance_lines_ggplot](#abundance_lines_ggplot) | create a ggplot object of the abundance data as a line graph
[network_phyloseq](#network_phyloseq) | creates a network of the co-occurrence of taxa
[nmds_phyloseq_ggplot](#nmds_phyloseq_ggplot)  | create a ggplot object of the NMDS from a phyloseq object
[phylogeny_bars_ggplot](#phylogeny_bars_ggplot) | create a ggplot barplot object of the compositons of each sample at a taxonomic level
[taxa_abundance_bars_ggplot](#taxa_abundance_bars_ggplot) | create a ggplot object of the abundance of taxa in each sample
[tsne_phyloseq_ggplot](#tsne_phyloseq_ggplot)  | create a ggplot object of the t-SNE from a phyloseq object

## Calculations
Call			 | Use
---------------- | ------------------------------------------------
[bootstrap_rho](#bootstrap_rho) | runs permutations of the otu_table to calculate a significant $\rho$ value
[co_occurrence](#co_occurrence) | calculate co-occurrence between taxa
[curate_cooccurrence](#curate_cooccurrence) | subsets the co-occurence table to specific taxa

## Datasets
Originally I had created 2 mock phyloseq objects (`mock_phyloseq` and `mock_phyloseq2`) that had no real-world data but served to show simple examples of how the functions worked. 

Then I decided that I should include a real example of microbiome data (`soil_column`) becasue it's always nice to see real examples. `soil_column` is a <a href="https://www.frontiersin.org/articles/10.3389/fmicb.2018.03197/full" target="_blank" >published dataset</a>  from my lab-group. The data is from an experiment where they looked at the microbial composition of farmland soil before and after manure application, over time, using 16S-sequencing.



