<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Schuyler D. Smith" />

<meta name="date" content="2019-05-10" />

<title>Graphs</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = false;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">phylosmith</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Overview
  </a>
</li>
<li>
  <a href="data_wrangling.html">
    <span class="fa fa-file"></span>
     
    Data Wrangling
  </a>
</li>
<li>
  <a href="graphs.html">
    <span class="fa fa-bar-chart"></span>
     
    Graphs
  </a>
</li>
<li>
  <a href="calculations.html">
    <span class="fa fa-calculator"></span>
     
    Calculations
  </a>
</li>
<li>
  <a href="https://github.com/schuyler-smith/phylosmith/tree/master/R">
    <span class="fa fa-code"></span>
     
    Source Code
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="http://github.com/schuyler-smith">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://germslab.org/">
    <span class="fa fa-flask"></span>
     
  </a>
</li>
<li>
  <a href="https://www.bcb.iastate.edu/people/schuyler-smith">
    <span class="fa fa-university"></span>
     
  </a>
</li>
<li>
  <a href="mailto:sdsmith@iastate.edu">
    <span class="fa fa-envelope fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Graphs</h1>
<h4 class="author"><a href="https://schuyler-smith.github.io/" target="_blank" >Schuyler D. Smith</a></h4>
<h4 class="date">2019-05-10</h4>

</div>


<p><br></p>
<div id="abundance_heatmap_ggplot" class="section level2">
<h2>abundance_heatmap_ggplot</h2>
<p><a name="abundance_heatmap_ggplot"></a></p>
<p>Create a ggplot object heatmap of abundances of each taxa for each sample.</p>
<p><br> <strong><em>Usage</em></strong></p>
<pre class="r"><code>abundance_heatmap_ggplot(phyloseq_obj, classification = NULL, treatment, subset = NULL, transformation = &#39;none&#39;, colors = &#39;default&#39;)</code></pre>
<p><br> <strong><em>Arguments</em></strong></p>
<table>
<colgroup>
<col width="25%" />
<col width="75%" />
</colgroup>
<thead>
<tr class="header">
<th>Call</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>phyloseq_obj</code></td>
<td>A phyloseq-class object. It must contain <code>sample_data</code> with information about each sample, and it must contain <code>tax_table</code> with information about each taxa/gene.</td>
</tr>
<tr class="even">
<td><code>treatment</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the <code>sample_data</code>. This can be a <code>vector</code> of multiple columns and they will be combined into a new column.</td>
</tr>
<tr class="odd">
<td><code>subset</code></td>
<td>A factor within the <code>treatment</code>. This will remove any samples that to not contain this factor. This can be a vector of multiple factors to subset on.</td>
</tr>
<tr class="even">
<td><code>classification</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the <code>tax_table</code> for the factor to use for node colors.</td>
</tr>
<tr class="odd">
<td><code>transformation</code></td>
<td>Transformation to be used on the data. “none”, “relative_abundance”, “log”, “log10”, “log1p”, “log2”, “asn”, “atanh”, “boxcox”, “exp”, “identity”, “logit”, “probability”, “probit”, “reciprocal”, “reverse” and “sqrt”.</td>
</tr>
<tr class="even">
<td><code>colors</code></td>
<td>Name of a color set from the <a href="https://cran.r-project.org/web/packages/RColorBrewer/RColorBrewer.pdf" target="_blank" ><code>RColorBrewer</code></a> package or a vector palete of R-accepted colors.</td>
</tr>
</tbody>
</table>
<p><strong><em>Examples</em></strong></p>
<pre class="r"><code>abundance_heatmap_ggplot(soil_column, classification = &#39;phylum&#39;, treatment = c(&#39;Matrix&#39;, &#39;Treatment&#39;), subset = &#39;Soil&#39;, transformation = &#39;log10&#39;)</code></pre>
<p><img src="graphs_files/figure-html/abundance_heatmap_ggplot-1.png" width="1536" /></p>
<p><br></p>
<hr />
<p><br></p>
</div>
<div id="abundance_lines_ggplot" class="section level2">
<h2>abundance_lines_ggplot</h2>
<p><a name="abundance_lines_ggplot"></a></p>
<p>Create a ggplot object line graph of abundances across samples</p>
<p><br> <strong><em>Usage</em></strong></p>
<pre class="r"><code>abundance_lines_ggplot(phyloseq_obj, classification, treatment, subset = NULL, relative_abundance = FALSE, points = TRUE, colors = &#39;default&#39;)</code></pre>
<p><br> <strong><em>Arguments</em></strong></p>
<table>
<colgroup>
<col width="25%" />
<col width="75%" />
</colgroup>
<thead>
<tr class="header">
<th>Call</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>phyloseq_obj</code></td>
<td>A phyloseq-class object. It must contain <code>sample_data</code> with information about each sample, and it must contain <code>tax_table</code> with information about each taxa/gene.</td>
</tr>
<tr class="even">
<td><code>classification</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the <code>tax_table</code> for the factor to use for node colors.</td>
</tr>
<tr class="odd">
<td><code>treatment</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the sample_data. This can be a vector of multiple columns and they will be combined into a new column.</td>
</tr>
<tr class="even">
<td><code>subset</code></td>
<td>A factor within the <code>treatment</code>. This will remove any samples that to not contain this factor. This can be a vector of multiple factors to subset on.</td>
</tr>
<tr class="odd">
<td><code>relative_abundance</code></td>
<td>If <code>TRUE</code>, transforms the abundance data into relative abundance by sample.</td>
</tr>
<tr class="even">
<td><code>points</code></td>
<td>Whether or not to display point on the lines for each data-point.</td>
</tr>
<tr class="odd">
<td><code>colors</code></td>
<td>Name of a color set from the <a href="https://cran.r-project.org/web/packages/RColorBrewer/RColorBrewer.pdf" target="_blank" ><code>RColorBrewer</code></a> package or a vector palete of R-accepted colors.</td>
</tr>
</tbody>
</table>
<p><strong><em>Examples</em></strong></p>
<pre class="r"><code>abundance_lines_ggplot(conglomerate_samples(soil_column, treatment = c(&#39;Matrix&#39;, &#39;Treatment&#39;), merge_on = &#39;Day&#39;), classification = &#39;phylum&#39;, treatment = c(&#39;Matrix&#39;, &#39;Treatment&#39;), subset = &#39;Soil&#39;, relative_abundance = TRUE)</code></pre>
<p><img src="graphs_files/figure-html/abundance_lines_ggplot-1.png" width="1536" /></p>
<p><br></p>
<hr />
<p><br></p>
</div>
<div id="network_ps" class="section level2">
<h2>network_ps</h2>
<p><a name="network_ps"></a></p>
<p>Create a ggplot object of the co-occurrence network.</p>
<p><br> <strong><em>Usage</em></strong></p>
<pre class="r"><code>network_ps(phyloseq_obj, treatment = NULL, subset = NULL, co_occurrence_table = NULL, classification = NULL, node_colors = &#39;default&#39;, cluster = FALSE, cluster_colors = &#39;default&#39;, buffer = 0.5)</code></pre>
<p><br> <strong><em>Arguments</em></strong></p>
<table>
<colgroup>
<col width="25%" />
<col width="75%" />
</colgroup>
<thead>
<tr class="header">
<th>Call</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>phyloseq_obj</code></td>
<td>A phyloseq-class object. It must contain <code>sample_data</code> with information about each sample, and it must contain <code>tax_table</code> with information about each taxa/gene.</td>
</tr>
<tr class="even">
<td><code>treatment</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the sample_data. This can be a vector of multiple columns and they will be combined into a new column.</td>
</tr>
<tr class="odd">
<td><code>subset</code></td>
<td>A factor within the treatment. This will remove any samples that to not contain this factor. This can be a vector of multiple factors to subset on.</td>
</tr>
<tr class="even">
<td><code>co_occurrence_table</code></td>
<td>Table of the co-occurrence of taxa/genes in the phyloseq_obj, computed using <code>co_occurrence()</code>. If no table is given, it will be computed with the phyloseq_obj, using the given treatment and p = 0.05 with a rho cut-off of 0.8.</td>
</tr>
<tr class="odd">
<td><code>classification</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the <code>tax_table</code> for the factor to use for node colors.</td>
</tr>
<tr class="even">
<td><code>node_colors</code></td>
<td>Name of a color set from the <a href="https://cran.r-project.org/web/packages/RColorBrewer/RColorBrewer.pdf" target="_blank" ><code>RColorBrewer</code></a> package or a vector palete of R-accepted colors.</td>
</tr>
<tr class="odd">
<td><code>cluster</code></td>
<td>if <code>TRUE</code>, will use <code>igraph</code>’s <code>cluster_fast_greedy</code> method. Alternatively, you may pass a vector of cluster assignments with order corresponding to the order of the taxa_names in the <code>phyloseq_obj</code>.</td>
</tr>
<tr class="even">
<td><code>cluster_colors</code></td>
<td>Name of a color set from the <a href="https://cran.r-project.org/web/packages/RColorBrewer/RColorBrewer.pdf" target="_blank" ><code>RColorBrewer</code></a> package or a vector palete of R-accepted colors.</td>
</tr>
<tr class="odd">
<td><code>buffer</code></td>
<td>Amount of space beyond the points to extend the cluster (aesthetic argument).</td>
</tr>
</tbody>
</table>
<p><strong><em>Examples</em></strong></p>
<pre class="r"><code>network_ps(soil_column, treatment = c(&#39;Matrix&#39;, &#39;Treatment&#39;), subset = &#39;Soil Manure&#39;, co_occurrence_table = NULL, classification = &#39;phylum&#39;)</code></pre>
<p><img src="graphs_files/figure-html/network_ps-1.png" width="1536" /></p>
<p><br></p>
<hr />
<p><br></p>
</div>
<div id="network_layout_ps" class="section level2">
<h2>network_layout_ps</h2>
<p><a name="network_layout_ps"></a></p>
<p>Create an layout_igraph object of the co-occurrence from a phyloseq object. This can be input into the network_ps function, or used for other network creating scripts. The purpose is to be able to create reproducible and comparable graphics.</p>
<p><br> <strong><em>Usage</em></strong></p>
<pre class="r"><code>network_layout_ps(phyloseq_obj, classification = NULL, treatment = NULL, subset = NULL, co_occurrence_table = NULL, algorithm = &#39;fr&#39;)</code></pre>
<p><br> <strong><em>Arguments</em></strong></p>
<table>
<colgroup>
<col width="25%" />
<col width="75%" />
</colgroup>
<thead>
<tr class="header">
<th>Call</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>phyloseq_obj</code></td>
<td>A phyloseq-class object. It must contain <code>sample_data</code> with information about each sample, and it must contain <code>tax_table</code> with information about each taxa/gene.</td>
</tr>
<tr class="even">
<td><code>treatment</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the sample_data. This can be a vector of multiple columns and they will be combined into a new column.</td>
</tr>
<tr class="odd">
<td><code>subset</code></td>
<td>A factor within the treatment. This will remove any samples that to not contain this factor. This can be a vector of multiple factors to subset on.</td>
</tr>
<tr class="even">
<td><code>co_occurrence_table</code></td>
<td>Table of the co-occurrence of taxa/genes in the phyloseq_obj, computed using <code>co_occurrence()</code>. If no table is given, it will be computed with the phyloseq_obj, using the given treatment and p = 0.05 with a rho cut-off of 0.8.</td>
</tr>
<tr class="odd">
<td><code>classification</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the <code>tax_table</code> for the factor to use for node colors.</td>
</tr>
<tr class="even">
<td><code>algorithm</code></td>
<td>Supported <code>igraph::layout_</code> algorithm.</td>
</tr>
</tbody>
</table>
<p><strong><em>Examples</em></strong></p>
<pre class="r"><code>network_layout_ps(soil_column, treatment = c(&#39;Matrix&#39;, &#39;Treatment&#39;), subset = &#39;Soil Manure&#39;, algorithm = &#39;kk&#39;)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["x"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["y"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["name"],"name":[3],"type":["fctr"],"align":["left"]},{"label":["ggraph.orig_index"],"name":[4],"type":["int"],"align":["right"]},{"label":["circular"],"name":[5],"type":["lgl"],"align":["right"]},{"label":["ggraph.index"],"name":[6],"type":["int"],"align":["right"]}],"data":[{"1":"1.380706420","2":"4.1102338","3":"OTU_5","4":"1","5":"FALSE","6":"1"},{"1":"0.609678638","2":"1.9460259","3":"OTU_6","4":"2","5":"FALSE","6":"2"},{"1":"3.412974289","2":"5.2898697","3":"OTU_8","4":"3","5":"FALSE","6":"3"},{"1":"5.240338699","2":"-7.6571953","3":"OTU_18","4":"4","5":"FALSE","6":"4"},{"1":"5.861015833","2":"-9.0174825","3":"OTU_33","4":"5","5":"FALSE","6":"5"},{"1":"4.625254400","2":"-10.5054384","3":"OTU_38","4":"6","5":"FALSE","6":"6"},{"1":"7.665669570","2":"-8.3382574","3":"OTU_39","4":"7","5":"FALSE","6":"7"},{"1":"4.856560099","2":"-9.3355045","3":"OTU_43","4":"8","5":"FALSE","6":"8"},{"1":"4.940426181","2":"-7.1490726","3":"OTU_45","4":"9","5":"FALSE","6":"9"},{"1":"6.815016745","2":"-9.7404849","3":"OTU_46","4":"10","5":"FALSE","6":"10"},{"1":"-5.440315308","2":"-2.9588859","3":"OTU_48","4":"11","5":"FALSE","6":"11"},{"1":"-3.892436122","2":"3.5256149","3":"OTU_51","4":"12","5":"FALSE","6":"12"},{"1":"7.716091320","2":"2.9315826","3":"OTU_52","4":"13","5":"FALSE","6":"13"},{"1":"3.805954109","2":"-6.1467015","3":"OTU_53","4":"14","5":"FALSE","6":"14"},{"1":"10.478048146","2":"-4.1108874","3":"OTU_55","4":"15","5":"FALSE","6":"15"},{"1":"2.060966941","2":"-8.8010390","3":"OTU_56","4":"16","5":"FALSE","6":"16"},{"1":"6.883577910","2":"-7.5350439","3":"OTU_57","4":"17","5":"FALSE","6":"17"},{"1":"6.371602307","2":"-8.6536158","3":"OTU_61","4":"18","5":"FALSE","6":"18"},{"1":"4.415283384","2":"-7.6206133","3":"OTU_66","4":"19","5":"FALSE","6":"19"},{"1":"6.793981560","2":"-7.6412232","3":"OTU_70","4":"20","5":"FALSE","6":"20"},{"1":"-1.544040089","2":"3.8475701","3":"OTU_72","4":"21","5":"FALSE","6":"21"},{"1":"10.350139663","2":"-3.6993681","3":"OTU_73","4":"22","5":"FALSE","6":"22"},{"1":"-0.049671303","2":"2.7115581","3":"OTU_74","4":"23","5":"FALSE","6":"23"},{"1":"-2.391284140","2":"2.8302584","3":"OTU_79","4":"24","5":"FALSE","6":"24"},{"1":"9.294368793","2":"-3.3404664","3":"OTU_82","4":"25","5":"FALSE","6":"25"},{"1":"-0.646933113","2":"2.1375802","3":"OTU_86","4":"26","5":"FALSE","6":"26"},{"1":"9.568992224","2":"4.9648345","3":"OTU_87","4":"27","5":"FALSE","6":"27"},{"1":"12.288752159","2":"4.1518264","3":"OTU_89","4":"28","5":"FALSE","6":"28"},{"1":"0.866053964","2":"4.7937038","3":"OTU_108","4":"29","5":"FALSE","6":"29"},{"1":"9.790646931","2":"7.4583860","3":"OTU_134","4":"30","5":"FALSE","6":"30"},{"1":"9.189900548","2":"6.2118757","3":"OTU_144","4":"31","5":"FALSE","6":"31"},{"1":"3.421183544","2":"-10.7545559","3":"OTU_146","4":"32","5":"FALSE","6":"32"},{"1":"-4.440756977","2":"-8.2342251","3":"OTU_160","4":"33","5":"FALSE","6":"33"},{"1":"-5.778906469","2":"-8.3516667","3":"OTU_164","4":"34","5":"FALSE","6":"34"},{"1":"6.134602776","2":"-8.6731426","3":"OTU_166","4":"35","5":"FALSE","6":"35"},{"1":"-1.784813296","2":"6.9326221","3":"OTU_174","4":"36","5":"FALSE","6":"36"},{"1":"7.189734014","2":"10.0898766","3":"OTU_176","4":"37","5":"FALSE","6":"37"},{"1":"7.328727856","2":"-6.9315037","3":"OTU_177","4":"38","5":"FALSE","6":"38"},{"1":"-8.529404684","2":"-4.2325627","3":"OTU_179","4":"39","5":"FALSE","6":"39"},{"1":"6.745897561","2":"6.2788743","3":"OTU_181","4":"40","5":"FALSE","6":"40"},{"1":"7.511410658","2":"-8.1472381","3":"OTU_183","4":"41","5":"FALSE","6":"41"},{"1":"1.346988105","2":"-7.8509668","3":"OTU_187","4":"42","5":"FALSE","6":"42"},{"1":"7.744198797","2":"-5.2874570","3":"OTU_190","4":"43","5":"FALSE","6":"43"},{"1":"-2.394526048","2":"6.3713330","3":"OTU_194","4":"44","5":"FALSE","6":"44"},{"1":"-1.223852185","2":"3.3374412","3":"OTU_197","4":"45","5":"FALSE","6":"45"},{"1":"-0.662631362","2":"3.3941991","3":"OTU_198","4":"46","5":"FALSE","6":"46"},{"1":"-1.604826350","2":"2.5067239","3":"OTU_204","4":"47","5":"FALSE","6":"47"},{"1":"2.329224431","2":"5.2803975","3":"OTU_206","4":"48","5":"FALSE","6":"48"},{"1":"0.621580389","2":"2.9084569","3":"OTU_208","4":"49","5":"FALSE","6":"49"},{"1":"9.825978538","2":"-2.0681908","3":"OTU_209","4":"50","5":"FALSE","6":"50"},{"1":"2.422476152","2":"6.7441881","3":"OTU_212","4":"51","5":"FALSE","6":"51"},{"1":"0.692229676","2":"7.3798656","3":"OTU_217","4":"52","5":"FALSE","6":"52"},{"1":"-3.642480875","2":"3.6755055","3":"OTU_220","4":"53","5":"FALSE","6":"53"},{"1":"-2.071278328","2":"1.2485240","3":"OTU_255","4":"54","5":"FALSE","6":"54"},{"1":"-0.172001963","2":"1.1113122","3":"OTU_263","4":"55","5":"FALSE","6":"55"},{"1":"4.082268170","2":"-9.8295116","3":"OTU_283","4":"56","5":"FALSE","6":"56"},{"1":"9.293513024","2":"-6.7966199","3":"OTU_308","4":"57","5":"FALSE","6":"57"},{"1":"10.093847872","2":"-7.9362499","3":"OTU_309","4":"58","5":"FALSE","6":"58"},{"1":"6.896964189","2":"-9.1034181","3":"OTU_310","4":"59","5":"FALSE","6":"59"},{"1":"-5.355164476","2":"-9.2054555","3":"OTU_321","4":"60","5":"FALSE","6":"60"},{"1":"-5.961754915","2":"9.8628661","3":"OTU_327","4":"61","5":"FALSE","6":"61"},{"1":"-7.022570361","2":"8.9187883","3":"OTU_329","4":"62","5":"FALSE","6":"62"},{"1":"3.765825447","2":"-9.8297665","3":"OTU_330","4":"63","5":"FALSE","6":"63"},{"1":"-0.059595501","2":"9.4964008","3":"OTU_331","4":"64","5":"FALSE","6":"64"},{"1":"-2.313285006","2":"0.7613234","3":"OTU_341","4":"65","5":"FALSE","6":"65"},{"1":"-1.015282750","2":"5.1816950","3":"OTU_343","4":"66","5":"FALSE","6":"66"},{"1":"-8.995933854","2":"-3.4734774","3":"OTU_346","4":"67","5":"FALSE","6":"67"},{"1":"4.805590343","2":"1.6069444","3":"OTU_386","4":"68","5":"FALSE","6":"68"},{"1":"-1.592305470","2":"4.4990153","3":"OTU_391","4":"69","5":"FALSE","6":"69"},{"1":"11.032386346","2":"7.6024074","3":"OTU_394","4":"70","5":"FALSE","6":"70"},{"1":"6.810059948","2":"11.0609297","3":"OTU_399","4":"71","5":"FALSE","6":"71"},{"1":"-2.416670661","2":"3.9167893","3":"OTU_403","4":"72","5":"FALSE","6":"72"},{"1":"9.179624988","2":"8.8094655","3":"OTU_405","4":"73","5":"FALSE","6":"73"},{"1":"-0.857362813","2":"1.8405926","3":"OTU_406","4":"74","5":"FALSE","6":"74"},{"1":"10.042672202","2":"6.6890384","3":"OTU_408","4":"75","5":"FALSE","6":"75"},{"1":"1.881120590","2":"7.3219047","3":"OTU_412","4":"76","5":"FALSE","6":"76"},{"1":"-4.114156280","2":"2.9996354","3":"OTU_422","4":"77","5":"FALSE","6":"77"},{"1":"-2.954620949","2":"6.6955451","3":"OTU_430","4":"78","5":"FALSE","6":"78"},{"1":"-0.375080520","2":"5.0453802","3":"OTU_437","4":"79","5":"FALSE","6":"79"},{"1":"11.419362945","2":"6.0874792","3":"OTU_441","4":"80","5":"FALSE","6":"80"},{"1":"2.090367035","2":"1.4894088","3":"OTU_449","4":"81","5":"FALSE","6":"81"},{"1":"3.197397840","2":"-0.4866816","3":"OTU_450","4":"82","5":"FALSE","6":"82"},{"1":"2.769622868","2":"8.7186508","3":"OTU_465","4":"83","5":"FALSE","6":"83"},{"1":"0.319345685","2":"3.6467354","3":"OTU_467","4":"84","5":"FALSE","6":"84"},{"1":"-0.081058248","2":"8.2114787","3":"OTU_468","4":"85","5":"FALSE","6":"85"},{"1":"11.924558070","2":"6.4009252","3":"OTU_474","4":"86","5":"FALSE","6":"86"},{"1":"9.720084804","2":"8.4265960","3":"OTU_476","4":"87","5":"FALSE","6":"87"},{"1":"-0.632090360","2":"6.5830411","3":"OTU_480","4":"88","5":"FALSE","6":"88"},{"1":"-1.306545488","2":"6.2091811","3":"OTU_483","4":"89","5":"FALSE","6":"89"},{"1":"6.768984006","2":"-5.2786984","3":"OTU_499","4":"90","5":"FALSE","6":"90"},{"1":"-4.958277663","2":"-5.9261669","3":"OTU_502","4":"91","5":"FALSE","6":"91"},{"1":"8.014658858","2":"-7.3373859","3":"OTU_505","4":"92","5":"FALSE","6":"92"},{"1":"5.403132646","2":"-8.0058611","3":"OTU_509","4":"93","5":"FALSE","6":"93"},{"1":"11.151722968","2":"3.0480035","3":"OTU_512","4":"94","5":"FALSE","6":"94"},{"1":"-1.156608692","2":"2.6048231","3":"OTU_517","4":"95","5":"FALSE","6":"95"},{"1":"1.778163113","2":"-7.4728090","3":"OTU_519","4":"96","5":"FALSE","6":"96"},{"1":"0.465909803","2":"-9.9248730","3":"OTU_522","4":"97","5":"FALSE","6":"97"},{"1":"-0.761287015","2":"3.2855115","3":"OTU_528","4":"98","5":"FALSE","6":"98"},{"1":"0.386249164","2":"-9.3032185","3":"OTU_530","4":"99","5":"FALSE","6":"99"},{"1":"0.903809308","2":"-11.3766240","3":"OTU_532","4":"100","5":"FALSE","6":"100"},{"1":"5.299947745","2":"-6.3313956","3":"OTU_533","4":"101","5":"FALSE","6":"101"},{"1":"-2.674247150","2":"2.2505563","3":"OTU_535","4":"102","5":"FALSE","6":"102"},{"1":"-9.758068512","2":"-3.8277061","3":"OTU_551","4":"103","5":"FALSE","6":"103"},{"1":"9.888934719","2":"-5.4144367","3":"OTU_560","4":"104","5":"FALSE","6":"104"},{"1":"-8.450001464","2":"-6.5662266","3":"OTU_563","4":"105","5":"FALSE","6":"105"},{"1":"9.568018625","2":"3.7858789","3":"OTU_566","4":"106","5":"FALSE","6":"106"},{"1":"8.216243305","2":"-10.6029731","3":"OTU_587","4":"107","5":"FALSE","6":"107"},{"1":"12.368609963","2":"2.7181053","3":"OTU_611","4":"108","5":"FALSE","6":"108"},{"1":"10.512091338","2":"4.0898590","3":"OTU_625","4":"109","5":"FALSE","6":"109"},{"1":"1.022292246","2":"5.4447059","3":"OTU_627","4":"110","5":"FALSE","6":"110"},{"1":"9.131580827","2":"-2.6765502","3":"OTU_647","4":"111","5":"FALSE","6":"111"},{"1":"2.681547602","2":"-7.4040736","3":"OTU_651","4":"112","5":"FALSE","6":"112"},{"1":"-2.565698745","2":"4.4401017","3":"OTU_660","4":"113","5":"FALSE","6":"113"},{"1":"-0.016439323","2":"3.9817017","3":"OTU_661","4":"114","5":"FALSE","6":"114"},{"1":"-0.284566593","2":"2.3981578","3":"OTU_664","4":"115","5":"FALSE","6":"115"},{"1":"7.597778038","2":"-5.3490279","3":"OTU_726","4":"116","5":"FALSE","6":"116"},{"1":"6.449811212","2":"-4.5992587","3":"OTU_729","4":"117","5":"FALSE","6":"117"},{"1":"-1.852704421","2":"7.7168246","3":"OTU_739","4":"118","5":"FALSE","6":"118"},{"1":"1.501575886","2":"6.9689141","3":"OTU_740","4":"119","5":"FALSE","6":"119"},{"1":"-0.426948138","2":"-8.3414496","3":"OTU_755","4":"120","5":"FALSE","6":"120"},{"1":"4.396560670","2":"-8.3166029","3":"OTU_781","4":"121","5":"FALSE","6":"121"},{"1":"-1.287925378","2":"4.7977631","3":"OTU_796","4":"122","5":"FALSE","6":"122"},{"1":"-0.871937586","2":"-0.1890561","3":"OTU_842","4":"123","5":"FALSE","6":"123"},{"1":"-8.975732585","2":"-5.6887399","3":"OTU_864","4":"124","5":"FALSE","6":"124"},{"1":"12.940614286","2":"1.9125797","3":"OTU_874","4":"125","5":"FALSE","6":"125"},{"1":"10.683349202","2":"-4.7825550","3":"OTU_878","4":"126","5":"FALSE","6":"126"},{"1":"7.083393230","2":"-6.0400505","3":"OTU_880","4":"127","5":"FALSE","6":"127"},{"1":"-1.622625553","2":"1.6451570","3":"OTU_888","4":"128","5":"FALSE","6":"128"},{"1":"-0.735191647","2":"5.9748305","3":"OTU_891","4":"129","5":"FALSE","6":"129"},{"1":"0.189420994","2":"5.7862767","3":"OTU_894","4":"130","5":"FALSE","6":"130"},{"1":"4.360957995","2":"4.7865164","3":"OTU_895","4":"131","5":"FALSE","6":"131"},{"1":"-7.195072390","2":"-6.7873216","3":"OTU_901","4":"132","5":"FALSE","6":"132"},{"1":"9.305521743","2":"-8.5036070","3":"OTU_902","4":"133","5":"FALSE","6":"133"},{"1":"11.358869128","2":"3.4963882","3":"OTU_904","4":"134","5":"FALSE","6":"134"},{"1":"2.674299815","2":"-7.1440795","3":"OTU_906","4":"135","5":"FALSE","6":"135"},{"1":"-7.412983889","2":"-5.1235045","3":"OTU_916","4":"136","5":"FALSE","6":"136"},{"1":"6.323279491","2":"-6.8525624","3":"OTU_919","4":"137","5":"FALSE","6":"137"},{"1":"-1.013619704","2":"7.2405341","3":"OTU_9","4":"138","5":"FALSE","6":"138"},{"1":"6.743789187","2":"-10.3039434","3":"OTU_40","4":"139","5":"FALSE","6":"139"},{"1":"-0.491272763","2":"7.0911114","3":"OTU_126","4":"140","5":"FALSE","6":"140"},{"1":"-3.254919058","2":"8.6687901","3":"OTU_215","4":"141","5":"FALSE","6":"141"},{"1":"-2.405874807","2":"2.3862440","3":"OTU_221","4":"142","5":"FALSE","6":"142"},{"1":"0.266796926","2":"2.2240189","3":"OTU_243","4":"143","5":"FALSE","6":"143"},{"1":"-0.435252047","2":"2.7862921","3":"OTU_246","4":"144","5":"FALSE","6":"144"},{"1":"0.003480422","2":"0.6341327","3":"OTU_247","4":"145","5":"FALSE","6":"145"},{"1":"-8.374192548","2":"-4.7450233","3":"OTU_249","4":"146","5":"FALSE","6":"146"},{"1":"-2.844494821","2":"2.6135950","3":"OTU_251","4":"147","5":"FALSE","6":"147"},{"1":"3.210789408","2":"1.4599176","3":"OTU_259","4":"148","5":"FALSE","6":"148"},{"1":"-7.898623813","2":"5.3823382","3":"OTU_260","4":"149","5":"FALSE","6":"149"},{"1":"-1.270550260","2":"-0.6440361","3":"OTU_289","4":"150","5":"FALSE","6":"150"},{"1":"-3.122929085","2":"2.1423580","3":"OTU_294","4":"151","5":"FALSE","6":"151"},{"1":"-0.490037680","2":"1.4871169","3":"OTU_295","4":"152","5":"FALSE","6":"152"},{"1":"-1.044070079","2":"4.0863305","3":"OTU_400","4":"153","5":"FALSE","6":"153"},{"1":"9.430237556","2":"-5.6653480","3":"OTU_516","4":"154","5":"FALSE","6":"154"},{"1":"9.487592395","2":"-7.8153209","3":"OTU_534","4":"155","5":"FALSE","6":"155"},{"1":"-1.348932366","2":"-0.4052589","3":"OTU_538","4":"156","5":"FALSE","6":"156"},{"1":"-1.788922716","2":"1.5612814","3":"OTU_545","4":"157","5":"FALSE","6":"157"},{"1":"-2.105811326","2":"4.0010903","3":"OTU_555","4":"158","5":"FALSE","6":"158"},{"1":"-1.349156437","2":"0.4235805","3":"OTU_556","4":"159","5":"FALSE","6":"159"},{"1":"3.415746440","2":"3.0726443","3":"OTU_559","4":"160","5":"FALSE","6":"160"},{"1":"-1.194131453","2":"2.2159337","3":"OTU_571","4":"161","5":"FALSE","6":"161"},{"1":"1.152156315","2":"2.8001008","3":"OTU_575","4":"162","5":"FALSE","6":"162"},{"1":"-7.789813850","2":"-0.3467974","3":"OTU_588","4":"163","5":"FALSE","6":"163"},{"1":"3.207743594","2":"1.7831986","3":"OTU_600","4":"164","5":"FALSE","6":"164"},{"1":"1.201616587","2":"0.7643985","3":"OTU_648","4":"165","5":"FALSE","6":"165"},{"1":"-2.478839792","2":"3.1570947","3":"OTU_649","4":"166","5":"FALSE","6":"166"},{"1":"1.831139069","2":"2.6259600","3":"OTU_653","4":"167","5":"FALSE","6":"167"},{"1":"-0.384171309","2":"4.6665868","3":"OTU_655","4":"168","5":"FALSE","6":"168"},{"1":"6.472013413","2":"-5.6453425","3":"OTU_657","4":"169","5":"FALSE","6":"169"},{"1":"0.697457823","2":"-0.2283394","3":"OTU_658","4":"170","5":"FALSE","6":"170"},{"1":"-3.758449821","2":"-0.7725722","3":"OTU_659","4":"171","5":"FALSE","6":"171"},{"1":"-0.263475749","2":"0.7804606","3":"OTU_663","4":"172","5":"FALSE","6":"172"},{"1":"0.699478736","2":"3.5853169","3":"OTU_702","4":"173","5":"FALSE","6":"173"},{"1":"1.138551326","2":"5.2359918","3":"OTU_746","4":"174","5":"FALSE","6":"174"},{"1":"-6.181935457","2":"-7.3997275","3":"OTU_823","4":"175","5":"FALSE","6":"175"},{"1":"-4.141256248","2":"4.9258385","3":"OTU_830","4":"176","5":"FALSE","6":"176"},{"1":"0.819250665","2":"-7.2409280","3":"OTU_834","4":"177","5":"FALSE","6":"177"},{"1":"-3.411193700","2":"1.1998258","3":"OTU_872","4":"178","5":"FALSE","6":"178"},{"1":"13.827820139","2":"1.8164317","3":"OTU_11","4":"179","5":"FALSE","6":"179"},{"1":"8.830346263","2":"-2.6389274","3":"OTU_58","4":"180","5":"FALSE","6":"180"},{"1":"0.036838809","2":"-7.5665055","3":"OTU_97","4":"181","5":"FALSE","6":"181"},{"1":"0.764675525","2":"2.4424526","3":"OTU_244","4":"182","5":"FALSE","6":"182"},{"1":"-8.371708914","2":"1.1794672","3":"OTU_252","4":"183","5":"FALSE","6":"183"},{"1":"12.483915165","2":"0.2186423","3":"OTU_261","4":"184","5":"FALSE","6":"184"},{"1":"-0.173988333","2":"-7.8593592","3":"OTU_506","4":"185","5":"FALSE","6":"185"},{"1":"-5.651166307","2":"7.1847656","3":"OTU_673","4":"186","5":"FALSE","6":"186"},{"1":"3.516351553","2":"10.8412076","3":"OTU_836","4":"187","5":"FALSE","6":"187"},{"1":"-0.673731535","2":"10.2151469","3":"OTU_844","4":"188","5":"FALSE","6":"188"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p><br></p>
<hr />
<p><br></p>
</div>
<div id="nmds_phyloseq_ggplot" class="section level2">
<h2>nmds_phyloseq_ggplot</h2>
<p><a name="nmds_phyloseq_ggplot"></a></p>
<p>Creates a ggplot object of the NMDS ordination of samples.</p>
<p><br> <strong><em>Usage</em></strong></p>
<pre class="r"><code>nmds_phyloseq_ggplot(phyloseq_obj, treatment, circle = TRUE, labels = FALSE, colors = &#39;default&#39;, verbose = TRUE)</code></pre>
<p><br> <strong><em>Arguments</em></strong></p>
<table>
<colgroup>
<col width="25%" />
<col width="75%" />
</colgroup>
<thead>
<tr class="header">
<th>Call</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>phyloseq_obj</code></td>
<td>A phyloseq-class object. It must contain <code>sample_data</code> with information about each sample, and it must contain <code>tax_table</code> with information about each taxa/gene.</td>
</tr>
<tr class="even">
<td><code>treatment</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the sample_data. This can be a vector of multiple columns and they will be combined into a new column.</td>
</tr>
<tr class="odd">
<td><code>circle</code></td>
<td>Whether to add a stat_elipse around eache treatment factor on the graph.</td>
</tr>
<tr class="even">
<td><code>labels</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the sample_data to use to place labels of that factor instead of circle points.</td>
</tr>
<tr class="odd">
<td><code>colors</code></td>
<td>Name of a color set from the <a href="https://cran.r-project.org/web/packages/RColorBrewer/RColorBrewer.pdf" target="_blank" ><code>RColorBrewer</code></a> package or a vector palete of R-accepted colors.</td>
</tr>
<tr class="even">
<td><code>verbose</code></td>
<td>Whether or not to print the <code>vegan::metaMDS</code> stress convergence to console (TRUE) or not (FALSE).</td>
</tr>
</tbody>
</table>
<p><strong><em>Examples</em></strong></p>
<pre class="r"><code>nmds_phyloseq_ggplot(soil_column, c(&#39;Matrix&#39;, &#39;Treatment&#39;), circle = TRUE, verbose = FALSE)</code></pre>
<p><img src="graphs_files/figure-html/nmds_phyloseq_ggplot-1.png" width="1536" /></p>
<p><br></p>
<hr />
<p><br></p>
</div>
<div id="phylogeny_profile_ggplot" class="section level2">
<h2>phylogeny_profile_ggplot</h2>
<p><a name="phylogeny_profile_ggplot"></a></p>
<p>Creates a ggplot object of the stacked bar plots of a classification level in the tax_data, for each sample.</p>
<p><br> <strong><em>Usage</em></strong></p>
<pre class="r"><code>phylogeny_profile_ggplot(phyloseq_obj, classification, treatment, subset = NULL, merge = TRUE, relative_abundance = FALSE, colors = &#39;default&#39;)</code></pre>
<p><br> <strong><em>Arguments</em></strong></p>
<table>
<colgroup>
<col width="25%" />
<col width="75%" />
</colgroup>
<thead>
<tr class="header">
<th>Call</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>phyloseq_obj</code></td>
<td>A phyloseq-class object. It must contain <code>sample_data</code> with information about each sample, and it must contain <code>tax_table</code> with information about each taxa/gene.</td>
</tr>
<tr class="even">
<td><code>classification</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the <code>tax_table</code> for the factor to use for node colors.</td>
</tr>
<tr class="odd">
<td><code>treatment</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the sample_data. This can be a vector of multiple columns and they will be combined into a new column.</td>
</tr>
<tr class="even">
<td><code>subset</code></td>
<td>A factor within the treatment. This will remove any samples that to not contain this factor. This can be a vector of multiple factors to subset on.</td>
</tr>
<tr class="odd">
<td><code>merge</code></td>
<td>If <code>FALSE</code>, will show separation of individuals within each <code>classification</code>.</td>
</tr>
<tr class="even">
<td><code>relative_abundance</code></td>
<td>If <code>TRUE</code>, transforms the abundance data into relative abundance by sample.</td>
</tr>
<tr class="odd">
<td><code>colors</code></td>
<td>Name of a color set from the <a href="https://cran.r-project.org/web/packages/RColorBrewer/RColorBrewer.pdf" target="_blank" ><code>RColorBrewer</code></a> package or a vector palete of R-accepted colors.</td>
</tr>
</tbody>
</table>
<p><strong><em>Examples</em></strong></p>
<pre class="r"><code>phylogeny_profile_ggplot(soil_column, classification = &#39;phylum&#39;, treatment = c(&#39;Matrix&#39;, &#39;Treatment&#39;), subset = &#39;Soil&#39;, merge = TRUE, relative_abundance = TRUE)</code></pre>
<p><img src="graphs_files/figure-html/phylogeny_profile_ggplot-1.png" width="1536" /></p>
<p><br></p>
<hr />
<p><br></p>
</div>
<div id="taxa_abundance_bars_ggplot" class="section level2">
<h2>taxa_abundance_bars_ggplot</h2>
<p><a name="taxa_abundance_bars_ggplot"></a></p>
<p>Creates a ggplot object of the side-by-side bar plots of a classification level in the tax_data.</p>
<p><br> <strong><em>Usage</em></strong></p>
<pre class="r"><code>taxa_abundance_bars_ggplot(phyloseq_obj, classification = NULL, treatment, subset = NULL, transformation = &#39;none&#39;, colors = &#39;default&#39;)</code></pre>
<p><br> <strong><em>Arguments</em></strong></p>
<table>
<colgroup>
<col width="25%" />
<col width="75%" />
</colgroup>
<thead>
<tr class="header">
<th>Call</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>phyloseq_obj</code></td>
<td>A phyloseq-class object. It must contain <code>sample_data</code> with information about each sample, and it must contain tax_table() with information about each taxa/gene.</td>
</tr>
<tr class="even">
<td><code>classification</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the <code>tax_table</code> for the factor to use for node colors.</td>
</tr>
<tr class="odd">
<td><code>treatment</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the sample_data. This can be a vector of multiple columns and they will be combined into a new column.</td>
</tr>
<tr class="even">
<td><code>subset</code></td>
<td>A factor within the <code>treatment</code>. This will remove any samples that to not contain this factor. This can be a vector of multiple factors to subset on.</td>
</tr>
<tr class="odd">
<td><code>transformation</code></td>
<td>Transformation to be used on the data. “none”, “mean”, “median”, “sd”, “log”, “log10”</td>
</tr>
<tr class="even">
<td><code>colors</code></td>
<td>Name of a color set from the <a href="https://cran.r-project.org/web/packages/RColorBrewer/RColorBrewer.pdf" target="_blank" ><code>RColorBrewer</code></a> package or a vector palete of R-accepted colors.</td>
</tr>
</tbody>
</table>
<p><strong><em>Examples</em></strong></p>
<pre class="r"><code>taxa_abundance_bars_ggplot(taxa_filter(soil_column, frequency = 0.8), classification = &#39;phylum&#39;, treatment = c(&#39;Matrix&#39;, &#39;Treatment&#39;), subset = &#39;Control&#39;, transformation = &#39;mean&#39;)</code></pre>
<p><img src="graphs_files/figure-html/taxa_abundance_bars_ggplot-1.png" width="1536" /></p>
<p><br></p>
<hr />
<p><br></p>
</div>
<div id="tsne_phyloseq_ggplot" class="section level2">
<h2>tsne_phyloseq_ggplot</h2>
<p><a name="tsne_phyloseq_ggplot"></a></p>
<p>Creates a ggplot object of the t-SNE ordination of the samples. t-Distributed Tochastic Neighbor Embedding is a machine learning algorithm of nonlinear dimensionality reduction for high-dimensional data.</p>
<p><br> <strong><em>Usage</em></strong></p>
<pre class="r"><code>tsne_phyloseq_ggplot(phyloseq_obj, treatment, perplexity = 10, circle = TRUE, labels = FALSE, colors = &#39;default&#39;)</code></pre>
<p><br> <strong><em>Arguments</em></strong></p>
<table>
<colgroup>
<col width="25%" />
<col width="75%" />
</colgroup>
<thead>
<tr class="header">
<th>Call</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>phyloseq_obj</code></td>
<td>A phyloseq-class object. It must contain <code>sample_data</code> with information about each sample, and it must contain <code>tax_table</code> with information about each taxa/gene.</td>
</tr>
<tr class="even">
<td><code>treatment</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the <code>sample_data</code>. This can be a <code>vector</code> of multiple columns and they will be combined into a new column.</td>
</tr>
<tr class="odd">
<td><code>perplexity</code></td>
<td>similar to selecting the number of neighbors to consider in decision making (should not be bigger than 3 * perplexity &lt; nrow(X) - 1, see Rtsne for interpretation)</td>
</tr>
<tr class="even">
<td><code>circle</code></td>
<td>Whether to add a <code>stat_elipse</code> around each <code>treatment</code> factor on the graph.</td>
</tr>
<tr class="odd">
<td><code>labels</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the <code>sample_data</code> to use to place labels of that factor instead of circle points.</td>
</tr>
<tr class="even">
<td><code>colors</code></td>
<td>Name of a color set from the <a href="https://cran.r-project.org/web/packages/RColorBrewer/RColorBrewer.pdf" target="_blank" ><code>RColorBrewer</code></a> package or a vector palete of R-accepted colors.</td>
</tr>
</tbody>
</table>
<p><strong><em>Examples</em></strong></p>
<pre class="r"><code>tsne_phyloseq_ggplot(soil_column, treatment = c(&#39;Matrix&#39;, &#39;Treatment&#39;), perplexity = 10)</code></pre>
<p><img src="graphs_files/figure-html/tsne_phyloseq_ggplot-1.png" width="1536" /></p>
<p><br></p>
<hr />
<p><br></p>
</div>

<p><br>
<br>
<strong><a href="https://schuyler-smith.github.io/" target="_blank">Schuyler Smith</a></strong>
<br>
Ph.D. Student - Bioinformatics and Computational Biology<br>
Iowa State University.  Ames, IA.</p>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
