<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Schuyler D. Smith" />

<meta name="date" content="2019-07-08" />

<title>Graphs</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = false;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">phylosmith</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Overview
  </a>
</li>
<li>
  <a href="data_wrangling.html">
    <span class="fa fa-file"></span>
     
    Data Wrangling
  </a>
</li>
<li>
  <a href="graphs.html">
    <span class="fa fa-bar-chart"></span>
     
    Graphs
  </a>
</li>
<li>
  <a href="calculations.html">
    <span class="fa fa-calculator"></span>
     
    Calculations
  </a>
</li>
<li>
  <a href="https://github.com/schuyler-smith/phylosmith/tree/master/R">
    <span class="fa fa-code"></span>
     
    Source Code
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="http://github.com/schuyler-smith">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://germslab.org/">
    <span class="fa fa-flask"></span>
     
  </a>
</li>
<li>
  <a href="https://www.bcb.iastate.edu/people/schuyler-smith">
    <span class="fa fa-university"></span>
     
  </a>
</li>
<li>
  <a href="mailto:sdsmith@iastate.edu">
    <span class="fa fa-envelope fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Graphs</h1>
<h4 class="author"><a href="https://schuyler-smith.github.io/" target="_blank" >Schuyler D. Smith</a></h4>
<h4 class="date">2019-07-08</h4>

</div>


<p><br></p>
<div id="abundance_heatmap" class="section level2">
<h2>abundance_heatmap</h2>
<p><a name="abundance_heatmap"></a></p>
<p>Create a ggplot object heatmap of abundances of each taxa for each sample.</p>
<p><br> <strong><em>Usage</em></strong></p>
<pre class="r"><code>abundance_heatmap(phyloseq_obj, classification = NULL, treatment, subset = NULL, transformation = &#39;none&#39;, colors = &#39;default&#39;)</code></pre>
<p><br> <strong><em>Arguments</em></strong></p>
<table>
<colgroup>
<col width="25%" />
<col width="75%" />
</colgroup>
<thead>
<tr class="header">
<th>Call</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>phyloseq_obj</code></td>
<td>A phyloseq-class object. It must contain <code>sample_data</code> with information about each sample, and it must contain <code>tax_table</code> with information about each taxa/gene.</td>
</tr>
<tr class="even">
<td><code>treatment</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the <code>sample_data</code>. This can be a <code>vector</code> of multiple columns and they will be combined into a new column.</td>
</tr>
<tr class="odd">
<td><code>subset</code></td>
<td>A factor within the <code>treatment</code>. This will remove any samples that to not contain this factor. This can be a vector of multiple factors to subset on.</td>
</tr>
<tr class="even">
<td><code>classification</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the <code>tax_table</code> for the factor to use for node colors.</td>
</tr>
<tr class="odd">
<td><code>transformation</code></td>
<td>Transformation to be used on the data. “none”, “relative_abundance”, “log”, “log10”, “log1p”, “log2”, “asn”, “atanh”, “boxcox”, “exp”, “identity”, “logit”, “probability”, “probit”, “reciprocal”, “reverse” and “sqrt”.</td>
</tr>
<tr class="even">
<td><code>colors</code></td>
<td>Name of a color set from the <a href="https://cran.r-project.org/web/packages/RColorBrewer/RColorBrewer.pdf" target="_blank" ><code>RColorBrewer</code></a> package or a vector palete of R-accepted colors.</td>
</tr>
</tbody>
</table>
<p><strong><em>Examples</em></strong></p>
<pre class="r"><code>abundance_heatmap(derep_sc, classification = &#39;Phylum&#39;, treatment = c(&#39;Matrix&#39;, &#39;Treatment&#39;), subset = &#39;Soil&#39;, transformation = &#39;log10&#39;)</code></pre>
<p><img src="graphs_files/figure-html/abundance_heatmap-1.png" width="1536" /></p>
<p><br></p>
<hr />
<p><br></p>
</div>
<div id="abundance_lines" class="section level2">
<h2>abundance_lines</h2>
<p><a name="abundance_lines"></a></p>
<p>Create a ggplot object line graph of abundances across samples</p>
<p><br> <strong><em>Usage</em></strong></p>
<pre class="r"><code>abundance_lines(phyloseq_obj, classification, treatment, subset = NULL, relative_abundance = FALSE, points = TRUE, colors = &#39;default&#39;)</code></pre>
<p><br> <strong><em>Arguments</em></strong></p>
<table>
<colgroup>
<col width="25%" />
<col width="75%" />
</colgroup>
<thead>
<tr class="header">
<th>Call</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>phyloseq_obj</code></td>
<td>A phyloseq-class object. It must contain <code>sample_data</code> with information about each sample, and it must contain <code>tax_table</code> with information about each taxa/gene.</td>
</tr>
<tr class="even">
<td><code>classification</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the <code>tax_table</code> for the factor to use for node colors.</td>
</tr>
<tr class="odd">
<td><code>treatment</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the sample_data. This can be a vector of multiple columns and they will be combined into a new column.</td>
</tr>
<tr class="even">
<td><code>subset</code></td>
<td>A factor within the <code>treatment</code>. This will remove any samples that to not contain this factor. This can be a vector of multiple factors to subset on.</td>
</tr>
<tr class="odd">
<td><code>relative_abundance</code></td>
<td>If <code>TRUE</code>, transforms the abundance data into relative abundance by sample.</td>
</tr>
<tr class="even">
<td><code>points</code></td>
<td>Whether or not to display point on the lines for each data-point.</td>
</tr>
<tr class="odd">
<td><code>colors</code></td>
<td>Name of a color set from the <a href="https://cran.r-project.org/web/packages/RColorBrewer/RColorBrewer.pdf" target="_blank" ><code>RColorBrewer</code></a> package or a vector palete of R-accepted colors.</td>
</tr>
</tbody>
</table>
<p><strong><em>Examples</em></strong></p>
<pre class="r"><code>abundance_lines(derep_sc, classification = &#39;Phylum&#39;, treatment = c(&#39;Matrix&#39;, &#39;Treatment&#39;), subset = &#39;Soil&#39;, relative_abundance = TRUE)</code></pre>
<p><img src="graphs_files/figure-html/abundance_lines-1.png" width="1536" /></p>
<p><br></p>
<hr />
<p><br></p>
</div>
<div id="co_occurrence_network" class="section level2">
<h2>co_occurrence_network</h2>
<p><a name="co_occurrence_network"></a></p>
<p>Create a ggplot object of the co-occurrence network.</p>
<p><br> <strong><em>Usage</em></strong></p>
<pre class="r"><code>co_occurrence_network(phyloseq_obj, treatment = NULL, subset = NULL, co_occurrence_table = NULL, classification = NULL, node_colors = &#39;default&#39;, cluster = FALSE, cluster_colors = &#39;default&#39;, buffer = 0.5)</code></pre>
<p><br> <strong><em>Arguments</em></strong></p>
<table>
<colgroup>
<col width="25%" />
<col width="75%" />
</colgroup>
<thead>
<tr class="header">
<th>Call</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>phyloseq_obj</code></td>
<td>A phyloseq-class object. It must contain <code>sample_data</code> with information about each sample, and it must contain <code>tax_table</code> with information about each taxa/gene.</td>
</tr>
<tr class="even">
<td><code>treatment</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the sample_data. This can be a vector of multiple columns and they will be combined into a new column.</td>
</tr>
<tr class="odd">
<td><code>subset</code></td>
<td>A factor within the treatment. This will remove any samples that to not contain this factor. This can be a vector of multiple factors to subset on.</td>
</tr>
<tr class="even">
<td><code>co_occurrence_table</code></td>
<td>Table of the co-occurrence of taxa/genes in the phyloseq_obj, computed using <code>co_occurrence()</code>. If no table is given, it will be computed with the phyloseq_obj, using the given treatment and p = 0.05 with a rho cut-off of 0.8.</td>
</tr>
<tr class="odd">
<td><code>classification</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the <code>tax_table</code> for the factor to use for node colors.</td>
</tr>
<tr class="even">
<td><code>node_colors</code></td>
<td>Name of a color set from the <a href="https://cran.r-project.org/web/packages/RColorBrewer/RColorBrewer.pdf" target="_blank" ><code>RColorBrewer</code></a> package or a vector palete of R-accepted colors.</td>
</tr>
<tr class="odd">
<td><code>cluster</code></td>
<td>if <code>TRUE</code>, will use <code>igraph</code>’s <code>cluster_fast_greedy</code> method. Alternatively, you may pass a vector of cluster assignments with order corresponding to the order of the taxa_names in the <code>phyloseq_obj</code>.</td>
</tr>
<tr class="even">
<td><code>cluster_colors</code></td>
<td>Name of a color set from the <a href="https://cran.r-project.org/web/packages/RColorBrewer/RColorBrewer.pdf" target="_blank" ><code>RColorBrewer</code></a> package or a vector palete of R-accepted colors.</td>
</tr>
<tr class="odd">
<td><code>buffer</code></td>
<td>Amount of space beyond the points to extend the cluster (aesthetic argument).</td>
</tr>
</tbody>
</table>
<p><strong><em>Examples</em></strong></p>
<pre class="r"><code>co_occurrence_network(soil_column, treatment = c(&#39;Matrix&#39;, &#39;Treatment&#39;), subset = &#39;Soil Amended&#39;, co_occurrence_table = NULL, classification = &#39;Phylum&#39;)</code></pre>
<p><img src="graphs_files/figure-html/co_occurrence_network-1.png" width="1536" /></p>
<p><br></p>
<hr />
<p><br></p>
</div>
<div id="network_layout_ps" class="section level2">
<h2>network_layout_ps</h2>
<p><a name="network_layout_ps"></a></p>
<p>Create an layout_igraph object of the co-occurrence from a phyloseq object. This can be input into the co_occurrence_network function, or used for other network creating scripts. The purpose is to be able to create reproducible and comparable graphics.</p>
<p><br> <strong><em>Usage</em></strong></p>
<pre class="r"><code>network_layout_ps(phyloseq_obj, classification = NULL, treatment = NULL, subset = NULL, co_occurrence_table = NULL, algorithm = &#39;fr&#39;)</code></pre>
<p><br> <strong><em>Arguments</em></strong></p>
<table>
<colgroup>
<col width="25%" />
<col width="75%" />
</colgroup>
<thead>
<tr class="header">
<th>Call</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>phyloseq_obj</code></td>
<td>A phyloseq-class object. It must contain <code>sample_data</code> with information about each sample, and it must contain <code>tax_table</code> with information about each taxa/gene.</td>
</tr>
<tr class="even">
<td><code>treatment</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the sample_data. This can be a vector of multiple columns and they will be combined into a new column.</td>
</tr>
<tr class="odd">
<td><code>subset</code></td>
<td>A factor within the treatment. This will remove any samples that to not contain this factor. This can be a vector of multiple factors to subset on.</td>
</tr>
<tr class="even">
<td><code>co_occurrence_table</code></td>
<td>Table of the co-occurrence of taxa/genes in the phyloseq_obj, computed using <code>co_occurrence()</code>. If no table is given, it will be computed with the phyloseq_obj, using the given treatment and p = 0.05 with a rho cut-off of 0.8.</td>
</tr>
<tr class="odd">
<td><code>classification</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the <code>tax_table</code> for the factor to use for node colors.</td>
</tr>
<tr class="even">
<td><code>algorithm</code></td>
<td>Supported <code>igraph::layout_</code> algorithm.</td>
</tr>
</tbody>
</table>
<p><strong><em>Examples</em></strong></p>
<pre class="r"><code>network_layout_ps(soil_column, treatment = c(&#39;Matrix&#39;, &#39;Treatment&#39;), subset = &#39;Soil Amended&#39;, algorithm = &#39;kk&#39;)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["x"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["y"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["name"],"name":[3],"type":["fctr"],"align":["left"]},{"label":["ggraph.orig_index"],"name":[4],"type":["int"],"align":["right"]},{"label":["circular"],"name":[5],"type":["lgl"],"align":["right"]},{"label":["ggraph.index"],"name":[6],"type":["int"],"align":["right"]}],"data":[{"1":"7.1715992","2":"3.70842738","3":"OTU_5","4":"1","5":"FALSE","6":"1"},{"1":"4.7123654","2":"4.22111151","3":"OTU_6","4":"2","5":"FALSE","6":"2"},{"1":"6.1543102","2":"7.78921962","3":"OTU_8","4":"3","5":"FALSE","6":"3"},{"1":"7.6194363","2":"-0.83312911","3":"OTU_9","4":"4","5":"FALSE","6":"4"},{"1":"-11.0051507","2":"1.12118658","3":"OTU_11","4":"5","5":"FALSE","6":"5"},{"1":"-4.9890435","2":"-9.64095592","3":"OTU_18","4":"6","5":"FALSE","6":"6"},{"1":"-12.0739246","2":"-2.54455862","3":"OTU_28","4":"7","5":"FALSE","6":"7"},{"1":"-4.2182345","2":"-11.09703548","3":"OTU_33","4":"8","5":"FALSE","6":"8"},{"1":"-4.4611419","2":"-6.53481795","3":"OTU_38","4":"9","5":"FALSE","6":"9"},{"1":"-3.3099059","2":"-8.70914348","3":"OTU_39","4":"10","5":"FALSE","6":"10"},{"1":"-1.5393351","2":"-9.59591453","3":"OTU_40","4":"11","5":"FALSE","6":"11"},{"1":"-1.5094696","2":"-10.55455189","3":"OTU_43","4":"12","5":"FALSE","6":"12"},{"1":"-2.7656013","2":"-10.51192759","3":"OTU_45","4":"13","5":"FALSE","6":"13"},{"1":"-4.3002561","2":"-8.32928566","3":"OTU_46","4":"14","5":"FALSE","6":"14"},{"1":"9.7553240","2":"-7.80268968","3":"OTU_48","4":"15","5":"FALSE","6":"15"},{"1":"-5.4681892","2":"-0.87330602","3":"OTU_51","4":"16","5":"FALSE","6":"16"},{"1":"-5.3691601","2":"1.13807764","3":"OTU_52","4":"17","5":"FALSE","6":"17"},{"1":"-7.5306662","2":"-8.76016311","3":"OTU_53","4":"18","5":"FALSE","6":"18"},{"1":"-9.5907656","2":"-7.76564215","3":"OTU_55","4":"19","5":"FALSE","6":"19"},{"1":"-6.8494586","2":"-5.53126573","3":"OTU_56","4":"20","5":"FALSE","6":"20"},{"1":"-4.8209113","2":"-10.47298863","3":"OTU_57","4":"21","5":"FALSE","6":"21"},{"1":"-2.6648137","2":"9.21261776","3":"OTU_58","4":"22","5":"FALSE","6":"22"},{"1":"-3.3693402","2":"-11.33177432","3":"OTU_61","4":"23","5":"FALSE","6":"23"},{"1":"-5.4063277","2":"-11.36843323","3":"OTU_66","4":"24","5":"FALSE","6":"24"},{"1":"-5.4857058","2":"-10.50058230","3":"OTU_70","4":"25","5":"FALSE","6":"25"},{"1":"6.3673634","2":"2.97058851","3":"OTU_72","4":"26","5":"FALSE","6":"26"},{"1":"0.7175892","2":"-13.05308735","3":"OTU_73","4":"27","5":"FALSE","6":"27"},{"1":"4.5070958","2":"3.57224930","3":"OTU_74","4":"28","5":"FALSE","6":"28"},{"1":"3.9348040","2":"1.70711542","3":"OTU_79","4":"29","5":"FALSE","6":"29"},{"1":"-5.0414700","2":"-4.82305081","3":"OTU_82","4":"30","5":"FALSE","6":"30"},{"1":"4.3368925","2":"3.13033025","3":"OTU_86","4":"31","5":"FALSE","6":"31"},{"1":"-6.9685080","2":"6.61359629","3":"OTU_87","4":"32","5":"FALSE","6":"32"},{"1":"-6.3738735","2":"8.46181504","3":"OTU_89","4":"33","5":"FALSE","6":"33"},{"1":"-4.7162874","2":"8.25594628","3":"OTU_97","4":"34","5":"FALSE","6":"34"},{"1":"7.2215782","2":"2.17503177","3":"OTU_108","4":"35","5":"FALSE","6":"35"},{"1":"9.0028911","2":"2.69254080","3":"OTU_126","4":"36","5":"FALSE","6":"36"},{"1":"-8.8140058","2":"6.96166572","3":"OTU_134","4":"37","5":"FALSE","6":"37"},{"1":"-8.1052579","2":"5.16220028","3":"OTU_144","4":"38","5":"FALSE","6":"38"},{"1":"-7.9716171","2":"-10.37900366","3":"OTU_146","4":"39","5":"FALSE","6":"39"},{"1":"3.1171883","2":"-10.31335347","3":"OTU_160","4":"40","5":"FALSE","6":"40"},{"1":"1.4920993","2":"-11.84603388","3":"OTU_164","4":"41","5":"FALSE","6":"41"},{"1":"-5.8256108","2":"-8.83325986","3":"OTU_166","4":"42","5":"FALSE","6":"42"},{"1":"9.1261429","2":"1.58155019","3":"OTU_174","4":"43","5":"FALSE","6":"43"},{"1":"6.5490611","2":"-10.66002519","3":"OTU_175","4":"44","5":"FALSE","6":"44"},{"1":"-8.7598857","2":"-1.88121184","3":"OTU_176","4":"45","5":"FALSE","6":"45"},{"1":"-5.8744333","2":"-9.78896018","3":"OTU_177","4":"46","5":"FALSE","6":"46"},{"1":"8.1502022","2":"-7.50491962","3":"OTU_179","4":"47","5":"FALSE","6":"47"},{"1":"-9.2498397","2":"3.53109128","3":"OTU_181","4":"48","5":"FALSE","6":"48"},{"1":"-3.5426712","2":"-10.68301608","3":"OTU_183","4":"49","5":"FALSE","6":"49"},{"1":"-9.4605039","2":"-8.35463355","3":"OTU_187","4":"50","5":"FALSE","6":"50"},{"1":"-5.1967336","2":"-7.76146661","3":"OTU_190","4":"51","5":"FALSE","6":"51"},{"1":"8.5069960","2":"0.79839924","3":"OTU_194","4":"52","5":"FALSE","6":"52"},{"1":"5.2233928","2":"2.14782409","3":"OTU_197","4":"53","5":"FALSE","6":"53"},{"1":"5.2676009","2":"2.72181294","3":"OTU_198","4":"54","5":"FALSE","6":"54"},{"1":"3.8781340","2":"1.22571114","3":"OTU_204","4":"55","5":"FALSE","6":"55"},{"1":"8.9886579","2":"1.96439135","3":"OTU_206","4":"56","5":"FALSE","6":"56"},{"1":"3.7476690","2":"2.43577518","3":"OTU_208","4":"57","5":"FALSE","6":"57"},{"1":"4.0104103","2":"-11.31441078","3":"OTU_209","4":"58","5":"FALSE","6":"58"},{"1":"4.2650308","2":"-2.10597085","3":"OTU_212","4":"59","5":"FALSE","6":"59"},{"1":"10.8223273","2":"0.48688644","3":"OTU_215","4":"60","5":"FALSE","6":"60"},{"1":"8.4150559","2":"-0.44136686","3":"OTU_217","4":"61","5":"FALSE","6":"61"},{"1":"7.4484676","2":"4.59407461","3":"OTU_220","4":"62","5":"FALSE","6":"62"},{"1":"6.2008052","2":"4.57011525","3":"OTU_221","4":"63","5":"FALSE","6":"63"},{"1":"2.9569348","2":"1.36936553","3":"OTU_243","4":"64","5":"FALSE","6":"64"},{"1":"-8.0796399","2":"6.38385516","3":"OTU_244","4":"65","5":"FALSE","6":"65"},{"1":"5.3306823","2":"3.17498541","3":"OTU_246","4":"66","5":"FALSE","6":"66"},{"1":"5.0713225","2":"5.37886923","3":"OTU_247","4":"67","5":"FALSE","6":"67"},{"1":"6.4388343","2":"-9.33878971","3":"OTU_249","4":"68","5":"FALSE","6":"68"},{"1":"2.6030135","2":"1.75789335","3":"OTU_251","4":"69","5":"FALSE","6":"69"},{"1":"-6.1503749","2":"3.94663461","3":"OTU_252","4":"70","5":"FALSE","6":"70"},{"1":"2.9616835","2":"2.62089577","3":"OTU_255","4":"71","5":"FALSE","6":"71"},{"1":"2.7913585","2":"-11.53626236","3":"OTU_256","4":"72","5":"FALSE","6":"72"},{"1":"0.9037360","2":"0.92993929","3":"OTU_259","4":"73","5":"FALSE","6":"73"},{"1":"-8.6056972","2":"1.97647291","3":"OTU_260","4":"74","5":"FALSE","6":"74"},{"1":"-11.7765276","2":"-0.77118806","3":"OTU_261","4":"75","5":"FALSE","6":"75"},{"1":"2.4181745","2":"2.50805152","3":"OTU_263","4":"76","5":"FALSE","6":"76"},{"1":"8.7405099","2":"-8.54313473","3":"OTU_283","4":"77","5":"FALSE","6":"77"},{"1":"1.1697503","2":"3.54579178","3":"OTU_289","4":"78","5":"FALSE","6":"78"},{"1":"5.4582147","2":"4.36112371","3":"OTU_294","4":"79","5":"FALSE","6":"79"},{"1":"3.6454326","2":"3.52608672","3":"OTU_295","4":"80","5":"FALSE","6":"80"},{"1":"-2.8878763","2":"-6.63610135","3":"OTU_308","4":"81","5":"FALSE","6":"81"},{"1":"-8.2635128","2":"-9.47599035","3":"OTU_309","4":"82","5":"FALSE","6":"82"},{"1":"-3.7068295","2":"-8.95521430","3":"OTU_310","4":"83","5":"FALSE","6":"83"},{"1":"9.1988366","2":"-4.73876126","3":"OTU_317","4":"84","5":"FALSE","6":"84"},{"1":"4.8572221","2":"-11.81651900","3":"OTU_321","4":"85","5":"FALSE","6":"85"},{"1":"-3.7691312","2":"8.81262811","3":"OTU_327","4":"86","5":"FALSE","6":"86"},{"1":"-2.1382042","2":"10.00163541","3":"OTU_329","4":"87","5":"FALSE","6":"87"},{"1":"-5.4485191","2":"-6.89261593","3":"OTU_330","4":"88","5":"FALSE","6":"88"},{"1":"9.8557352","2":"-1.93107384","3":"OTU_331","4":"89","5":"FALSE","6":"89"},{"1":"-11.1783468","2":"-5.84372490","3":"OTU_336","4":"90","5":"FALSE","6":"90"},{"1":"5.8458553","2":"5.41975552","3":"OTU_341","4":"91","5":"FALSE","6":"91"},{"1":"5.5511674","2":"0.21792662","3":"OTU_343","4":"92","5":"FALSE","6":"92"},{"1":"1.3462535","2":"-12.91654271","3":"OTU_344","4":"93","5":"FALSE","6":"93"},{"1":"7.7841204","2":"-9.43010253","3":"OTU_346","4":"94","5":"FALSE","6":"94"},{"1":"0.6816800","2":"6.31981419","3":"OTU_386","4":"95","5":"FALSE","6":"95"},{"1":"-7.7860694","2":"7.32874073","3":"OTU_387","4":"96","5":"FALSE","6":"96"},{"1":"5.3521173","2":"1.04222033","3":"OTU_391","4":"97","5":"FALSE","6":"97"},{"1":"-9.9339191","2":"0.77899430","3":"OTU_393","4":"98","5":"FALSE","6":"98"},{"1":"-11.7236076","2":"1.48371000","3":"OTU_394","4":"99","5":"FALSE","6":"99"},{"1":"-5.6317324","2":"8.41730349","3":"OTU_396","4":"100","5":"FALSE","6":"100"},{"1":"-6.7665325","2":"-1.51581507","3":"OTU_399","4":"101","5":"FALSE","6":"101"},{"1":"6.5983150","2":"2.42645786","3":"OTU_400","4":"102","5":"FALSE","6":"102"},{"1":"4.2506849","2":"0.62829194","3":"OTU_403","4":"103","5":"FALSE","6":"103"},{"1":"-11.6130541","2":"-4.10371273","3":"OTU_405","4":"104","5":"FALSE","6":"104"},{"1":"7.1558137","2":"-2.12245094","3":"OTU_406","4":"105","5":"FALSE","6":"105"},{"1":"-8.5282275","2":"0.70429370","3":"OTU_408","4":"106","5":"FALSE","6":"106"},{"1":"7.7128468","2":"-1.65402837","3":"OTU_412","4":"107","5":"FALSE","6":"107"},{"1":"-11.4900417","2":"-5.02913168","3":"OTU_417","4":"108","5":"FALSE","6":"108"},{"1":"-11.8635713","2":"-1.64736002","3":"OTU_419","4":"109","5":"FALSE","6":"109"},{"1":"8.3591356","2":"4.24244765","3":"OTU_422","4":"110","5":"FALSE","6":"110"},{"1":"-1.5539803","2":"10.08039672","3":"OTU_428","4":"111","5":"FALSE","6":"111"},{"1":"5.9687158","2":"-1.67691800","3":"OTU_430","4":"112","5":"FALSE","6":"112"},{"1":"-9.8047643","2":"-1.43134835","3":"OTU_432","4":"113","5":"FALSE","6":"113"},{"1":"6.8052216","2":"1.52653454","3":"OTU_437","4":"114","5":"FALSE","6":"114"},{"1":"-10.5343881","2":"-0.20643638","3":"OTU_441","4":"115","5":"FALSE","6":"115"},{"1":"0.5150038","2":"8.88195858","3":"OTU_444","4":"116","5":"FALSE","6":"116"},{"1":"3.4944559","2":"5.48424174","3":"OTU_449","4":"117","5":"FALSE","6":"117"},{"1":"1.8308191","2":"7.19300414","3":"OTU_450","4":"118","5":"FALSE","6":"118"},{"1":"-5.0270867","2":"6.91493073","3":"OTU_457","4":"119","5":"FALSE","6":"119"},{"1":"-5.8740330","2":"6.01027914","3":"OTU_461","4":"120","5":"FALSE","6":"120"},{"1":"8.6786032","2":"-3.08643604","3":"OTU_465","4":"121","5":"FALSE","6":"121"},{"1":"4.7371339","2":"1.81400864","3":"OTU_467","4":"122","5":"FALSE","6":"122"},{"1":"9.3447560","2":"-0.42299418","3":"OTU_468","4":"123","5":"FALSE","6":"123"},{"1":"-7.7880859","2":"2.88644204","3":"OTU_474","4":"124","5":"FALSE","6":"124"},{"1":"-10.9064837","2":"-2.04979795","3":"OTU_476","4":"125","5":"FALSE","6":"125"},{"1":"-3.1489883","2":"9.84352835","3":"OTU_478","4":"126","5":"FALSE","6":"126"},{"1":"3.8928008","2":"6.55684667","3":"OTU_480","4":"127","5":"FALSE","6":"127"},{"1":"6.4374087","2":"-0.52363161","3":"OTU_483","4":"128","5":"FALSE","6":"128"},{"1":"-11.9145660","2":"0.34490007","3":"OTU_487","4":"129","5":"FALSE","6":"129"},{"1":"7.5724239","2":"-8.48921892","3":"OTU_494","4":"130","5":"FALSE","6":"130"},{"1":"-4.7902733","2":"-12.55722756","3":"OTU_499","4":"131","5":"FALSE","6":"131"},{"1":"8.6894872","2":"-9.43040166","3":"OTU_502","4":"132","5":"FALSE","6":"132"},{"1":"-2.7369868","2":"-9.32963235","3":"OTU_505","4":"133","5":"FALSE","6":"133"},{"1":"2.1841234","2":"-12.70509363","3":"OTU_506","4":"134","5":"FALSE","6":"134"},{"1":"-3.7450776","2":"-11.84315067","3":"OTU_509","4":"135","5":"FALSE","6":"135"},{"1":"-7.1804553","2":"1.65079324","3":"OTU_512","4":"136","5":"FALSE","6":"136"},{"1":"-1.9452318","2":"-12.93346753","3":"OTU_516","4":"137","5":"FALSE","6":"137"},{"1":"-6.2898310","2":"0.16784121","3":"OTU_517","4":"138","5":"FALSE","6":"138"},{"1":"-8.1050531","2":"-7.29750484","3":"OTU_519","4":"139","5":"FALSE","6":"139"},{"1":"-8.7624967","2":"-4.63619464","3":"OTU_522","4":"140","5":"FALSE","6":"140"},{"1":"-8.2837742","2":"-0.67116020","3":"OTU_528","4":"141","5":"FALSE","6":"141"},{"1":"-9.7387486","2":"-6.15422519","3":"OTU_530","4":"142","5":"FALSE","6":"142"},{"1":"-10.6739220","2":"-4.51404153","3":"OTU_532","4":"143","5":"FALSE","6":"143"},{"1":"-2.3463255","2":"-11.65470798","3":"OTU_533","4":"144","5":"FALSE","6":"144"},{"1":"-2.0547712","2":"-7.22811707","3":"OTU_534","4":"145","5":"FALSE","6":"145"},{"1":"-10.5770391","2":"2.25162256","3":"OTU_535","4":"146","5":"FALSE","6":"146"},{"1":"1.7355935","2":"4.01988110","3":"OTU_538","4":"147","5":"FALSE","6":"147"},{"1":"6.3429255","2":"5.29350307","3":"OTU_545","4":"148","5":"FALSE","6":"148"},{"1":"6.0368458","2":"-8.00531487","3":"OTU_551","4":"149","5":"FALSE","6":"149"},{"1":"7.0362771","2":"2.67804377","3":"OTU_555","4":"150","5":"FALSE","6":"150"},{"1":"3.0222461","2":"4.38465887","3":"OTU_556","4":"151","5":"FALSE","6":"151"},{"1":"4.9796863","2":"6.63669311","3":"OTU_559","4":"152","5":"FALSE","6":"152"},{"1":"-8.0721160","2":"-8.19137691","3":"OTU_560","4":"153","5":"FALSE","6":"153"},{"1":"5.2070202","2":"-9.80260828","3":"OTU_563","4":"154","5":"FALSE","6":"154"},{"1":"-6.0898218","2":"4.85978131","3":"OTU_566","4":"155","5":"FALSE","6":"155"},{"1":"4.9905009","2":"3.49363465","3":"OTU_571","4":"156","5":"FALSE","6":"156"},{"1":"6.7793638","2":"4.09194894","3":"OTU_575","4":"157","5":"FALSE","6":"157"},{"1":"-1.3186081","2":"-7.81021318","3":"OTU_587","4":"158","5":"FALSE","6":"158"},{"1":"-10.4392250","2":"3.50365593","3":"OTU_588","4":"159","5":"FALSE","6":"159"},{"1":"2.7089519","2":"6.07424667","3":"OTU_600","4":"160","5":"FALSE","6":"160"},{"1":"5.7182147","2":"-10.75989712","3":"OTU_601","4":"161","5":"FALSE","6":"161"},{"1":"4.5687794","2":"3.22460775","3":"OTU_602","4":"162","5":"FALSE","6":"162"},{"1":"-0.8506786","2":"10.33250233","3":"OTU_604","4":"163","5":"FALSE","6":"163"},{"1":"-4.4282186","2":"5.94609018","3":"OTU_611","4":"164","5":"FALSE","6":"164"},{"1":"-7.0546566","2":"7.86620975","3":"OTU_621","4":"165","5":"FALSE","6":"165"},{"1":"-5.5961968","2":"3.01287842","3":"OTU_625","4":"166","5":"FALSE","6":"166"},{"1":"10.5672488","2":"-6.22919918","3":"OTU_627","4":"167","5":"FALSE","6":"167"},{"1":"-0.7417658","2":"-4.74210615","3":"OTU_647","4":"168","5":"FALSE","6":"168"},{"1":"2.4954745","2":"4.04861725","3":"OTU_648","4":"169","5":"FALSE","6":"169"},{"1":"4.4364799","2":"1.04370379","3":"OTU_649","4":"170","5":"FALSE","6":"170"},{"1":"-1.5259523","2":"-5.72546726","3":"OTU_651","4":"171","5":"FALSE","6":"171"},{"1":"6.6410031","2":"4.87588787","3":"OTU_653","4":"172","5":"FALSE","6":"172"},{"1":"6.2804650","2":"1.38807304","3":"OTU_655","4":"173","5":"FALSE","6":"173"},{"1":"-1.5674688","2":"-11.04111154","3":"OTU_657","4":"174","5":"FALSE","6":"174"},{"1":"2.2028198","2":"4.71046118","3":"OTU_658","4":"175","5":"FALSE","6":"175"},{"1":"4.6573357","2":"7.74597099","3":"OTU_659","4":"176","5":"FALSE","6":"176"},{"1":"4.4524715","2":"-0.21081070","3":"OTU_660","4":"177","5":"FALSE","6":"177"},{"1":"5.7160562","2":"2.27573510","3":"OTU_661","4":"178","5":"FALSE","6":"178"},{"1":"4.2469699","2":"4.82976229","3":"OTU_663","4":"179","5":"FALSE","6":"179"},{"1":"4.2730535","2":"2.78467100","3":"OTU_664","4":"180","5":"FALSE","6":"180"},{"1":"-8.2997657","2":"4.23300410","3":"OTU_673","4":"181","5":"FALSE","6":"181"},{"1":"2.8750045","2":"-8.94426941","3":"OTU_680","4":"182","5":"FALSE","6":"182"},{"1":"5.0336329","2":"-8.65206062","3":"OTU_683","4":"183","5":"FALSE","6":"183"},{"1":"-9.2100849","2":"5.38143887","3":"OTU_702","4":"184","5":"FALSE","6":"184"},{"1":"3.8292179","2":"-12.34457528","3":"OTU_704","4":"185","5":"FALSE","6":"185"},{"1":"2.1064453","2":"-10.47793645","3":"OTU_712","4":"186","5":"FALSE","6":"186"},{"1":"-6.4176624","2":"-4.71629813","3":"OTU_726","4":"187","5":"FALSE","6":"187"},{"1":"-6.7095379","2":"-7.24693024","3":"OTU_729","4":"188","5":"FALSE","6":"188"},{"1":"6.1212436","2":"-2.41703148","3":"OTU_739","4":"189","5":"FALSE","6":"189"},{"1":"5.0840898","2":"-2.15785353","3":"OTU_740","4":"190","5":"FALSE","6":"190"},{"1":"7.8653232","2":"2.06795599","3":"OTU_746","4":"191","5":"FALSE","6":"191"},{"1":"-7.8642056","2":"-3.59757167","3":"OTU_755","4":"192","5":"FALSE","6":"192"},{"1":"-5.6215845","2":"-12.11839295","3":"OTU_781","4":"193","5":"FALSE","6":"193"},{"1":"-3.9830078","2":"9.59239905","3":"OTU_793","4":"194","5":"FALSE","6":"194"},{"1":"-11.5158677","2":"-3.31967789","3":"OTU_794","4":"195","5":"FALSE","6":"195"},{"1":"5.8524440","2":"0.95561484","3":"OTU_796","4":"196","5":"FALSE","6":"196"},{"1":"4.1892376","2":"-9.20825878","3":"OTU_823","4":"197","5":"FALSE","6":"197"},{"1":"3.2308042","2":"-1.41366297","3":"OTU_830","4":"198","5":"FALSE","6":"198"},{"1":"2.8855532","2":"-12.70472862","3":"OTU_834","4":"199","5":"FALSE","6":"199"},{"1":"-11.3588947","2":"2.61327738","3":"OTU_836","4":"200","5":"FALSE","6":"200"},{"1":"1.7606083","2":"1.32852100","3":"OTU_842","4":"201","5":"FALSE","6":"201"},{"1":"-10.3619650","2":"4.36613611","3":"OTU_844","4":"202","5":"FALSE","6":"202"},{"1":"5.6536286","2":"1.61560336","3":"OTU_862","4":"203","5":"FALSE","6":"203"},{"1":"9.1268310","2":"-7.75808594","3":"OTU_864","4":"204","5":"FALSE","6":"204"},{"1":"1.6987130","2":"2.96392537","3":"OTU_872","4":"205","5":"FALSE","6":"205"},{"1":"-3.6452145","2":"7.77993799","3":"OTU_874","4":"206","5":"FALSE","6":"206"},{"1":"10.0143178","2":"-6.92992038","3":"OTU_878","4":"207","5":"FALSE","6":"207"},{"1":"-7.1734202","2":"-11.21367324","3":"OTU_880","4":"208","5":"FALSE","6":"208"},{"1":"-4.9864293","2":"9.26636800","3":"OTU_883","4":"209","5":"FALSE","6":"209"},{"1":"3.2659765","2":"0.93966217","3":"OTU_888","4":"210","5":"FALSE","6":"210"},{"1":"7.4078967","2":"0.30049091","3":"OTU_891","4":"211","5":"FALSE","6":"211"},{"1":"-9.9424605","2":"-3.12776155","3":"OTU_892","4":"212","5":"FALSE","6":"212"},{"1":"6.7579068","2":"0.09886258","3":"OTU_894","4":"213","5":"FALSE","6":"213"},{"1":"2.1496943","2":"-2.30373232","3":"OTU_895","4":"214","5":"FALSE","6":"214"},{"1":"7.2301316","2":"-10.61482742","3":"OTU_901","4":"215","5":"FALSE","6":"215"},{"1":"-0.9486481","2":"-8.60755257","3":"OTU_902","4":"216","5":"FALSE","6":"216"},{"1":"-10.1022260","2":"5.31566823","3":"OTU_904","4":"217","5":"FALSE","6":"217"},{"1":"-8.2951811","2":"-6.11043850","3":"OTU_906","4":"218","5":"FALSE","6":"218"},{"1":"5.5456964","2":"-11.69641837","3":"OTU_916","4":"219","5":"FALSE","6":"219"},{"1":"-3.9247378","2":"-10.00360221","3":"OTU_919","4":"220","5":"FALSE","6":"220"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p><br></p>
<hr />
<p><br></p>
</div>
<div id="nmds_phyloseq" class="section level2">
<h2>nmds_phyloseq</h2>
<p><a name="nmds_phyloseq"></a></p>
<p>Creates a ggplot object of the NMDS ordination of samples.</p>
<p><br> <strong><em>Usage</em></strong></p>
<pre class="r"><code>nmds_phyloseq(phyloseq_obj, treatment, circle = TRUE, labels = FALSE, colors = &#39;default&#39;, verbose = TRUE)</code></pre>
<p><br> <strong><em>Arguments</em></strong></p>
<table>
<colgroup>
<col width="25%" />
<col width="75%" />
</colgroup>
<thead>
<tr class="header">
<th>Call</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>phyloseq_obj</code></td>
<td>A phyloseq-class object. It must contain <code>sample_data</code> with information about each sample, and it must contain <code>tax_table</code> with information about each taxa/gene.</td>
</tr>
<tr class="even">
<td><code>treatment</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the sample_data. This can be a vector of multiple columns and they will be combined into a new column.</td>
</tr>
<tr class="odd">
<td><code>circle</code></td>
<td>Whether to add a stat_elipse around eache treatment factor on the graph.</td>
</tr>
<tr class="even">
<td><code>labels</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the sample_data to use to place labels of that factor instead of circle points.</td>
</tr>
<tr class="odd">
<td><code>colors</code></td>
<td>Name of a color set from the <a href="https://cran.r-project.org/web/packages/RColorBrewer/RColorBrewer.pdf" target="_blank" ><code>RColorBrewer</code></a> package or a vector palete of R-accepted colors.</td>
</tr>
<tr class="even">
<td><code>verbose</code></td>
<td>Whether or not to print the <code>vegan::metaMDS</code> stress convergence to console (TRUE) or not (FALSE).</td>
</tr>
</tbody>
</table>
<p><strong><em>Examples</em></strong></p>
<pre class="r"><code>nmds_phyloseq(soil_column, c(&#39;Matrix&#39;, &#39;Treatment&#39;), circle = TRUE, verbose = FALSE)</code></pre>
<p><img src="graphs_files/figure-html/nmds_phyloseq-1.png" width="1536" /></p>
<p><br></p>
<hr />
<p><br></p>
</div>
<div id="phylogeny_profile" class="section level2">
<h2>phylogeny_profile</h2>
<p><a name="phylogeny_profile"></a></p>
<p>Creates a ggplot object of the stacked bar plots of a classification level in the tax_data, for each sample.</p>
<p><br> <strong><em>Usage</em></strong></p>
<pre class="r"><code>phylogeny_profile(phyloseq_obj, classification, treatment, subset = NULL, merge = TRUE, relative_abundance = FALSE, colors = &#39;default&#39;)</code></pre>
<p><br> <strong><em>Arguments</em></strong></p>
<table>
<colgroup>
<col width="25%" />
<col width="75%" />
</colgroup>
<thead>
<tr class="header">
<th>Call</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>phyloseq_obj</code></td>
<td>A phyloseq-class object. It must contain <code>sample_data</code> with information about each sample, and it must contain <code>tax_table</code> with information about each taxa/gene.</td>
</tr>
<tr class="even">
<td><code>classification</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the <code>tax_table</code> for the factor to use for node colors.</td>
</tr>
<tr class="odd">
<td><code>treatment</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the sample_data. This can be a vector of multiple columns and they will be combined into a new column.</td>
</tr>
<tr class="even">
<td><code>subset</code></td>
<td>A factor within the treatment. This will remove any samples that to not contain this factor. This can be a vector of multiple factors to subset on.</td>
</tr>
<tr class="odd">
<td><code>merge</code></td>
<td>If <code>FALSE</code>, will show separation of individuals within each <code>classification</code>.</td>
</tr>
<tr class="even">
<td><code>relative_abundance</code></td>
<td>If <code>TRUE</code>, transforms the abundance data into relative abundance by sample.</td>
</tr>
<tr class="odd">
<td><code>colors</code></td>
<td>Name of a color set from the <a href="https://cran.r-project.org/web/packages/RColorBrewer/RColorBrewer.pdf" target="_blank" ><code>RColorBrewer</code></a> package or a vector palete of R-accepted colors.</td>
</tr>
</tbody>
</table>
<p><strong><em>Examples</em></strong></p>
<pre class="r"><code>phylogeny_profile(derep_sc, classification = &#39;Phylum&#39;, treatment = c(&#39;Matrix&#39;, &#39;Treatment&#39;), subset = &#39;Soil&#39;, merge = TRUE, relative_abundance = TRUE)</code></pre>
<p><img src="graphs_files/figure-html/phylogeny_profile-1.png" width="1536" /></p>
<p><br></p>
<hr />
<p><br></p>
</div>
<div id="taxa_abundance_bars" class="section level2">
<h2>taxa_abundance_bars</h2>
<p><a name="taxa_abundance_bars"></a></p>
<p>Creates a ggplot object of the side-by-side bar plots of a classification level in the tax_data.</p>
<p><br> <strong><em>Usage</em></strong></p>
<pre class="r"><code>taxa_abundance_bars(phyloseq_obj, classification = NULL, treatment, subset = NULL, transformation = &#39;none&#39;, colors = &#39;default&#39;)</code></pre>
<p><br> <strong><em>Arguments</em></strong></p>
<table>
<colgroup>
<col width="25%" />
<col width="75%" />
</colgroup>
<thead>
<tr class="header">
<th>Call</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>phyloseq_obj</code></td>
<td>A phyloseq-class object. It must contain <code>sample_data</code> with information about each sample, and it must contain tax_table() with information about each taxa/gene.</td>
</tr>
<tr class="even">
<td><code>classification</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the <code>tax_table</code> for the factor to use for node colors.</td>
</tr>
<tr class="odd">
<td><code>treatment</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the sample_data. This can be a vector of multiple columns and they will be combined into a new column.</td>
</tr>
<tr class="even">
<td><code>subset</code></td>
<td>A factor within the <code>treatment</code>. This will remove any samples that to not contain this factor. This can be a vector of multiple factors to subset on.</td>
</tr>
<tr class="odd">
<td><code>transformation</code></td>
<td>Transformation to be used on the data. “none”, “mean”, “median”, “sd”, “log”, “log10”</td>
</tr>
<tr class="even">
<td><code>colors</code></td>
<td>Name of a color set from the <a href="https://cran.r-project.org/web/packages/RColorBrewer/RColorBrewer.pdf" target="_blank" ><code>RColorBrewer</code></a> package or a vector palete of R-accepted colors.</td>
</tr>
</tbody>
</table>
<p><strong><em>Examples</em></strong></p>
<pre class="r"><code>taxa_abundance_bars(taxa_filter(soil_column, frequency = 0.8), classification = &#39;Phylum&#39;, treatment = c(&#39;Matrix&#39;, &#39;Treatment&#39;), subset = &#39;Unamended&#39;, transformation = &#39;mean&#39;)</code></pre>
<p><img src="graphs_files/figure-html/taxa_abundance_bars-1.png" width="1536" /></p>
<p><br></p>
<hr />
<p><br></p>
</div>
<div id="taxa_core_graph" class="section level2">
<h2>taxa_core_graph</h2>
<p><a name="taxa_core_graph"></a></p>
<p>Create graph of the core taxa seen in phyloseq-object over a range of abundance and smaple-frequency values.</p>
<p><br> <strong><em>Usage</em></strong></p>
<pre class="r"><code>taxa_core_graph(phyloseq_obj, treatment = NULL, subset = NULL, frequencies = seq(0.1, 1, 0.1), abundance_thresholds = seq(0.01, 1, 0.01), colors = &#39;default&#39;)</code></pre>
<p><br> <strong><em>Arguments</em></strong></p>
<table>
<colgroup>
<col width="25%" />
<col width="75%" />
</colgroup>
<thead>
<tr class="header">
<th>Call</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>phyloseq_obj</code></td>
<td>A phyloseq-class object. It must contain <code>sample_data</code> with information about each sample, and it must contain tax_table() with information about each taxa/gene.</td>
</tr>
<tr class="even">
<td><code>treatment</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the sample_data. This can be a vector of multiple columns and they will be combined into a new column.</td>
</tr>
<tr class="odd">
<td><code>subset</code></td>
<td>A factor within the <code>treatment</code>. This will remove any samples that to not contain this factor. This can be a vector of multiple factors to subset on.</td>
</tr>
<tr class="even">
<td><code>frequencies</code></td>
<td>The range of proportions of samples the taxa are found in.</td>
</tr>
<tr class="odd">
<td><code>abundance_thresholds</code></td>
<td>The range of minimum relative abundances the taxa are found in for each sample.</td>
</tr>
<tr class="even">
<td><code>colors</code></td>
<td>Name of a color set from the <a href="https://cran.r-project.org/web/packages/RColorBrewer/RColorBrewer.pdf" target="_blank" ><code>RColorBrewer</code></a> package or a vector palete of R-accepted colors.</td>
</tr>
</tbody>
</table>
<p><strong><em>Examples</em></strong></p>
<pre class="r"><code>taxa_core_graph(soil_column, abundance_thresholds = seq(0.01, 0.25, 0.01))</code></pre>
<p><img src="graphs_files/figure-html/taxa_core_graph-1.png" width="1536" /> <br></p>
<hr />
<p><br></p>
</div>
<div id="tsne_phyloseq" class="section level2">
<h2>tsne_phyloseq</h2>
<p><a name="tsne_phyloseq"></a></p>
<p>Creates a ggplot object of the t-SNE ordination of the samples. t-Distributed Tochastic Neighbor Embedding is a machine learning algorithm of nonlinear dimensionality reduction for high-dimensional data.</p>
<p><br> <strong><em>Usage</em></strong></p>
<pre class="r"><code>tsne_phyloseq(phyloseq_obj, treatment, perplexity = 10, circle = TRUE, labels = FALSE, colors = &#39;default&#39;)</code></pre>
<p><br> <strong><em>Arguments</em></strong></p>
<table>
<colgroup>
<col width="25%" />
<col width="75%" />
</colgroup>
<thead>
<tr class="header">
<th>Call</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>phyloseq_obj</code></td>
<td>A phyloseq-class object. It must contain <code>sample_data</code> with information about each sample, and it must contain <code>tax_table</code> with information about each taxa/gene.</td>
</tr>
<tr class="even">
<td><code>treatment</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the <code>sample_data</code>. This can be a <code>vector</code> of multiple columns and they will be combined into a new column.</td>
</tr>
<tr class="odd">
<td><code>perplexity</code></td>
<td>similar to selecting the number of neighbors to consider in decision making (should not be bigger than 3 * perplexity &lt; nrow(X) - 1, see Rtsne for interpretation)</td>
</tr>
<tr class="even">
<td><code>circle</code></td>
<td>Whether to add a <code>stat_elipse</code> around each <code>treatment</code> factor on the graph.</td>
</tr>
<tr class="odd">
<td><code>labels</code></td>
<td>Column name as a <code>string</code> or <code>numeric</code> in the <code>sample_data</code> to use to place labels of that factor instead of circle points.</td>
</tr>
<tr class="even">
<td><code>colors</code></td>
<td>Name of a color set from the <a href="https://cran.r-project.org/web/packages/RColorBrewer/RColorBrewer.pdf" target="_blank" ><code>RColorBrewer</code></a> package or a vector palete of R-accepted colors.</td>
</tr>
</tbody>
</table>
<p><strong><em>Examples</em></strong></p>
<pre class="r"><code>tsne_phyloseq(soil_column, treatment = c(&#39;Matrix&#39;, &#39;Treatment&#39;), perplexity = 10)</code></pre>
<p><img src="graphs_files/figure-html/tsne_phyloseq-1.png" width="1536" /></p>
<p><br></p>
<hr />
<p><br></p>
</div>

<p><br>
<br>
<strong><a href="https://schuyler-smith.github.io/" target="_blank">Schuyler Smith</a></strong>
<br>
Ph.D. Student - Bioinformatics and Computational Biology<br>
Iowa State University.  Ames, IA.</p>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
