<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Functions to help analyze data as phyloseq objects • phylosmith</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Functions to help analyze data as phyloseq objects">
<meta property="og:description" content="Functions for analyzing phyloseq objects.">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">phylosmith</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="articles/phylosmith.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="contents col-md-9">

<div id="phylosmith" class="section level2">
<h2 class="hasAnchor">
<a href="#phylosmith" class="anchor"></a><code>phylosmith</code>
</h2>
<p><a href="https://schuyler-smith.github.io/phylosmith/">vignette</a></p>
<p><a href="https://github.com/schuyler-smith/phylosmith"><img src="https://travis-ci.org/schuyler-smith/phylosmith.svg?branch=master" alt="Travis Build Status"></a> <a href="http://joss.theoj.org/papers/4d4780ab16c487764ebe108fa6bcdc2c"><img src="http://joss.theoj.org/papers/4d4780ab16c487764ebe108fa6bcdc2c/status.svg" alt="status"></a></p>
<p>A conglomeration of functions that I have written, that I find useful, for analyzing phyloseq objects. Phyloseq objects are a great data-standard for microbiome and gene-expression data.</p>
</div>
<div id="installation" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1></div>
<div id="requirements" class="section level2">
<h2 class="hasAnchor">
<a href="#requirements" class="anchor"></a>Requirements</h2>
<div id="linux-systems" class="section level3">
<h3 class="hasAnchor">
<a href="#linux-systems" class="anchor"></a>Linux Systems</h3>
<p>For some Linux systems you may need to install the following two programs through your terminal.</p>
<p>Ubuntu example:</p>
<pre><code>sudo apt install libgdal-dev libudunits2-dev</code></pre>
<p>These programs are required by some dependencies and may not come in your default OS distribution.</p>
</div>
<div id="windows-systems" class="section level3">
<h3 class="hasAnchor">
<a href="#windows-systems" class="anchor"></a>Windows Systems</h3>
<p>if you are working on WINDOWS you likely need to install the CRAN program <a href="https://cran.r-project.org/bin/windows/Rtools/" target="_blank">Rtools</a>.When prompted, select <code>add rtools to system PATH</code>.</p>
</div>
<div id="r" class="section level3">
<h3 class="hasAnchor">
<a href="#r" class="anchor"></a>R</h3>
<p>phylosmith depends on the usage of the <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0061217">phyloseq package</a> released by <a href="https://github.com/joey711">Dr. Paul McMurdie</a>. The package is maintained on BioConductor, and can be installed through R using the following commands:</p>
<pre><code>if(!requireNamespace("BiocManager", quietly = TRUE)){
  install.packages("BiocManager")
} 
BiocManager::install("phyloseq")</code></pre>
<p>Additionally, the package imports a number of other packages to use their advanced functions. These packages may install with the phylosmith installation, but it is always best to install independently.</p>
<pre><code>install.packages(c("devtools", RcppEigen", "RcppParallel", "Rtsne", "ggforce", "units"))</code></pre>
</div>
</div>
<div id="phylosmith-1" class="section level2">
<h2 class="hasAnchor">
<a href="#phylosmith-1" class="anchor"></a>phylosmith</h2>
<p>The package is hosted on Github, and can be installed through R with:</p>
<pre><code><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">devtools::install_github('schuyler-smith/phylosmith')
library(phylosmith)</a></code></pre>
</div>
<div id="functions" class="section level2">
<h2 class="hasAnchor">
<a href="#functions" class="anchor"></a>Functions</h2>
<div id="data-wrangling" class="section level3">
<h3 class="hasAnchor">
<a href="#data-wrangling" class="anchor"></a><a href="https://schuyler-smith.github.io/phylosmith/data_wrangling.html">Data Wrangling</a>
</h3>
<table class="table">
<colgroup>
<col width="25%">
<col width="75%">
</colgroup>
<thead><tr class="header">
<th>Call</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><a href="https://schuyler-smith.github.io/phylosmith/data_wrangling.html#common_taxa">common_taxa</a></td>
<td>find taxa common to each treatment</td>
</tr>
<tr class="even">
<td><a href="https://schuyler-smith.github.io/phylosmith/data_wrangling.html#conglomerate_samples">conglomerate_samples</a></td>
<td>Merge samples based on common factor within sample_data</td>
</tr>
<tr class="odd">
<td><a href="https://schuyler-smith.github.io/phylosmith/data_wrangling.html#conglomerate_taxa">conglomerate_taxa</a></td>
<td>conglomerate taxa by sample on a given classification level</td>
</tr>
<tr class="even">
<td><a href="https://schuyler-smith.github.io/phylosmith/data_wrangling.html#melt_phyloseq">melt_phyloseq</a></td>
<td>Melt a phyloseq object into a data.table.</td>
</tr>
<tr class="odd">
<td><a href="https://schuyler-smith.github.io/phylosmith/data_wrangling.html#merge_treatments">merge_treatments</a></td>
<td>combines multiple columns in meta-data into a single column</td>
</tr>
<tr class="even">
<td><a href="https://schuyler-smith.github.io/phylosmith/data_wrangling.html#relative_abundance">relative_abundance</a></td>
<td>transform abundance data to relative abundance</td>
</tr>
<tr class="odd">
<td><a href="https://schuyler-smith.github.io/phylosmith/data_wrangling.html#set_sample_order">set_sample_order</a></td>
<td>sets the order of the samples of a phyloseq object</td>
</tr>
<tr class="even">
<td><a href="https://schuyler-smith.github.io/phylosmith/data_wrangling.html#set_treatment_levels">set_treatment_levels</a></td>
<td>sets the order of the factors in a sample_data column</td>
</tr>
<tr class="odd">
<td><a href="https://schuyler-smith.github.io/phylosmith/data_wrangling.html#taxa_proportions">taxa_proportions</a></td>
<td>computes the proportion of a taxa classification</td>
</tr>
<tr class="even">
<td><a href="https://schuyler-smith.github.io/phylosmith/data_wrangling.html#taxa_filter">taxa_filter</a></td>
<td>filter taxa by proportion of samples seen in</td>
</tr>
<tr class="odd">
<td><a href="https://schuyler-smith.github.io/phylosmith/data_wrangling.html#unique_taxa">unique_taxa</a></td>
<td>find taxa unique to each treatment</td>
</tr>
</tbody>
</table>
</div>
<div id="graphs" class="section level3">
<h3 class="hasAnchor">
<a href="#graphs" class="anchor"></a><a href="https://schuyler-smith.github.io/phylosmith/graphs.html">Graphs</a>
</h3>
<table class="table">
<colgroup>
<col width="25%">
<col width="75%">
</colgroup>
<thead><tr class="header">
<th>Call</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><a href="https://schuyler-smith.github.io/phylosmith/graphs.html#abundance_heatmap_ggplot">abundance_heatmap_ggplot</a></td>
<td>create a ggplot object of the heatmaps of the abundance table</td>
</tr>
<tr class="even">
<td><a href="https://schuyler-smith.github.io/phylosmith/graphs.html#abundance_lines_ggplot">abundance_lines_ggplot</a></td>
<td>create a ggplot object of the abundance data as a line graph</td>
</tr>
<tr class="odd">
<td><a href="https://schuyler-smith.github.io/phylosmith/graphs.html#network_ps">network_ps</a></td>
<td>creates a ggplot object of the co-occurrence network of taxa</td>
</tr>
<tr class="even">
<td><a href="https://schuyler-smith.github.io/phylosmith/graphs.html#network_layout_ps">network_layout_ps</a></td>
<td>create a layout object for a network</td>
</tr>
<tr class="odd">
<td><a href="https://schuyler-smith.github.io/phylosmith/graphs.html#nmds_phyloseq_ggplot">nmds_phyloseq_ggplot</a></td>
<td>create a ggplot object of the NMDS from a phyloseq object</td>
</tr>
<tr class="even">
<td><a href="https://schuyler-smith.github.io/phylosmith/graphs.html#phylogeny_profile_ggplot">phylogeny_profile_ggplot</a></td>
<td>create a ggplot barplot object of the compositions of each sample at a taxonomic level</td>
</tr>
<tr class="odd">
<td><a href="https://schuyler-smith.github.io/phylosmith/graphs.html#taxa_abundance_bars_ggplot">taxa_abundance_bars_ggplot</a></td>
<td>create a ggplot object of the abundance of taxa in each sample</td>
</tr>
<tr class="even">
<td><a href="https://schuyler-smith.github.io/phylosmith/graphs.html#tsne_phyloseq_ggplot">tsne_phyloseq_ggplot</a></td>
<td>create a ggplot object of the t-SNE from a phyloseq object</td>
</tr>
</tbody>
</table>
</div>
<div id="calculations" class="section level3">
<h3 class="hasAnchor">
<a href="#calculations" class="anchor"></a><a href="https://schuyler-smith.github.io/phylosmith/calculations.html">Calculations</a>
</h3>
<table class="table">
<colgroup>
<col width="25%">
<col width="75%">
</colgroup>
<thead><tr class="header">
<th>Call</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><a href="https://schuyler-smith.github.io/phylosmith/calculations.html#permute_rho">co_occurrence</a></td>
<td>calculate co-occurrence between taxa</td>
</tr>
<tr class="even">
<td><a href="https://schuyler-smith.github.io/phylosmith/calculations.html#co_occurrence">permute_rho</a></td>
<td>runs permutations of the otu_table to calculate a significant $\rho$ value</td>
</tr>
<tr class="odd">
<td><a href="https://schuyler-smith.github.io/phylosmith/calculations.html##histogram_permuted_rhos">histogram_permuted_rhos</a></td>
<td>Create a ggplot object of the distribution of $\rho$ values.</td>
</tr>
<tr class="even">
<td><a href="https://schuyler-smith.github.io/phylosmith/calculations.html##quantile_permuted_rhos">quantile_permuted_rhos</a></td>
<td>calculate quantiles for the permuted $\rho$ values from the Spearman-rank co-occurrence</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="datasets" class="section level2">
<h2 class="hasAnchor">
<a href="#datasets" class="anchor"></a>Datasets</h2>
<p>Originally I had created 2 mock phyloseq objects (<code>mock_phyloseq</code> and <code>mock_phyloseq2</code>) that had no real-world data but served to show simple examples of how the functions worked.</p>
<p>Then I decided that I should include a real example of microbiome data (<code>soil_column</code>) because it’s always nice to see real examples. <code>soil_column</code> is a <a href="https://www.frontiersin.org/articles/10.3389/fmicb.2018.03197/full" target="_blank">published dataset</a> from my lab-group. The data is from an experiment where they looked at the microbial composition of farmland soil before and after manure application, over time, using 16S-sequencing.</p>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Report a bug at <br><a href="https://github.com/schuyler-smith/phylosmith.issues">https://​github.com/​schuyler-smith/​phylosmith.issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Schuyler Smith <br><small class="roles"> Maintainer </small>  </li>
</ul>
</div>

  </div>
</div>

      <footer><div class="copyright">
  <p>Developed by Schuyler Smith.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
